### Co-mutation and clonality

```{r setup, eval=TRUE, include=FALSE}
library(knitr)
library(kableExtra)
opts_knit$set(root.dir = '/Users/bowmanr/Projects/scDNA')
```

Some of our motivations for this section were:
1) Are epigenetic mutations found in the same cell, if so, are these cells the dominant clone?
2) Are there Tet2 IDH in the same sample, and if so same cell?
3) Are there signaling in the same sample, and if so same cell?


Upon code review.. there are way more efficient ways of writing this. However this reproduces what is in the manuscript. The more efficient method is included in the appendix here, but it changes the order 
```{r,warning=FALSE,message=FALSE,fig.align="center",fig.width=4,fig.height=4}
options(stringsAsFactors = FALSE)
library(UpSetR)
library(tidyr)
library(dplyr)
library(RColorBrewer)
library(ggplot2)
library(pals)
library(cowplot)

setwd("/Users/bowmanr/Projects/scDNA")
final_sample_summary<-readRDS(file="./data/final_sample_summary.rds")
pheno<-readRDS(file="./data/pheno.rds")

genes_of_interest <- c("DNMT3A","TET2","IDH2","IDH1","ASXL1","FLT3","NRAS","KRAS","JAK2","PTPN11")
DTAI_genes <- c("DNMT3A","TET2","IDH2","IDH1","ASXL1")
signaling_genes <- c("FLT3","JAK2","NRAS","KRAS","PTPN11")
list_of_mutants<-sapply(genes_of_interest,function(gene){
                              do.call(c,(lapply(names(final_sample_summary),function(sample){
                                  if(any(grepl(gene,
                                               colnames(final_sample_summary[[sample]]$NGT)))){
                                    return(sample)
                                  }
                                })))
                    })

sample_data<-do.call(rbind,lapply(names(final_sample_summary),function(sample){
  data.frame("Sample"=sample,
            "DNMT3A"=ifelse(any(grepl("DNMT3A",colnames(final_sample_summary[[sample]]$NGT))),1,0),
             "TET2"=ifelse(any(grepl("TET2",colnames(final_sample_summary[[sample]]$NGT))),1,0),
             "IDH2"=ifelse(any(grepl("IDH2",colnames(final_sample_summary[[sample]]$NGT))),1,0),
             "IDH1"=ifelse(any(grepl("IDH1",colnames(final_sample_summary[[sample]]$NGT))),1,0),
             "ASXL1"=ifelse(any(grepl("ASXL1",colnames(final_sample_summary[[sample]]$NGT))),1,0),
             "FLT3"=ifelse(any(grepl("FLT3",colnames(final_sample_summary[[sample]]$NGT))),1,0),
             "JAK2"=ifelse(any(grepl("JAK2",colnames(final_sample_summary[[sample]]$NGT))),1,0),
             "NRAS"=ifelse(any(grepl("NRAS",colnames(final_sample_summary[[sample]]$NGT))),1,0),
             "KRAS"=ifelse(any(grepl("KRAS",colnames(final_sample_summary[[sample]]$NGT))),1,0),
             "PTPN11"=ifelse(any(grepl("PTPN11",colnames(final_sample_summary[[sample]]$NGT))),1,0)
            )
}))


dominant_clone_data<-do.call(rbind,lapply(names(final_sample_summary),function(sample){
  # select the clones
  clones<-final_sample_summary[[sample]]$Clones%>%select(Clone)
  # Extract the mutations
  mutations <- colnames(final_sample_summary[[sample]]$NGT%>%select(!Clone))
  # Identify the dominant clone, which is the last one in this data frame
  WT_clone_to_exclude <-!apply(do.call(rbind,strsplit(as.character(clones[,1]),split="_")),1,function(x){all(x=="0")})
  dominant_clone <- tail(clones[WT_clone_to_exclude,],n=1)
  # Separate the clone into a deduplicated NGT matrix
  dominant_variants<-final_sample_summary[[sample]]$Clones%>%
                              select(Clone)%>%
                              separate(col=Clone,into=mutations,sep="_",remove=FALSE)%>%
                              pivot_longer(cols=mutations,names_to="Variant",values_to="Genotype")%>%
                              filter(Clone==`dominant_clone`&Genotype!=0)%>%pull(Variant)
  
  return(data.frame("Sample"=sample,
             "DNMT3A"=ifelse(any(grepl("DNMT3A",dominant_variants)),1,0),
             "TET2"=ifelse(any(grepl("TET2",dominant_variants)),1,0),
             "IDH2"=ifelse(any(grepl("IDH2",dominant_variants)),1,0),
             "IDH1"=ifelse(any(grepl("IDH1",dominant_variants)),1,0),
             "ASXL1"=ifelse(any(grepl("ASXL1",dominant_variants)),1,0),
             "FLT3"=ifelse(any(grepl("FLT3",dominant_variants)),1,0),
             "JAK2"=ifelse(any(grepl("JAK2",dominant_variants)),1,0),
             "NRAS"=ifelse(any(grepl("NRAS",dominant_variants)),1,0),
             "KRAS"=ifelse(any(grepl("KRAS",dominant_variants)),1,0),
             "PTPN11"=ifelse(any(grepl("PTPN11",dominant_variants)),1,0)))
            }))

sample_data$Match<-ifelse(sapply(dominant_clone_data$Sample,function(sample){
  all(dominant_clone_data%>%filter(Sample==sample)%>%select(`DTAI_genes`)==
      sample_data%>%filter(Sample==sample)%>%select(`DTAI_genes`))
}),"Match","Absent")


test_set<-left_join(sample_data,pheno,by="Sample")
test_set$Match[is.na(test_set$Match)]<-"Absent"

Myfunc <- function(row,feature) {
  data <- (row[feature]=="Match")
}

AML<-upset(test_set%>%filter(grepl("AML|MPN",Dx)), 
           sets=DTAI_genes,
           order.by = c("degree"),
           main.bar.color = "grey60",decreasing=FALSE,
           mainbar.y.label = "Number of samples",
           sets.x.label = "Number of \n mutant samples",
           text.scale=1.25,
           shade.alpha = 0.75,keep.order=TRUE,
           show.numbers=FALSE,
           mb.ratio = c(0.6, 0.4),
           queries=list(list(query = Myfunc, params = list("Match"), color = brewer.pal(5,"Reds")[5], active = TRUE )))

AML
```

And now for CH only
```{r,warning=FALSE,message=FALSE,fig.align="center",fig.width=4,fig.height=4}
CH<-upset(test_set%>%filter(Dx=="CH"), sets=DTAI_genes,order.by = c("degree"),
           main.bar.color = "grey60",decreasing=FALSE,
           mainbar.y.label = "Number of samples",
           sets.x.label = "Number of \n mutant samples",
           text.scale=1.25,
           shade.alpha = 0.75,
           show.numbers=FALSE,
           queries=list(list(query = Myfunc, params = list("Match"), color = brewer.pal(5,"Reds")[5], active = TRUE )))

CH
```

### Clone level network grapbs

```{r fig.align="center", warning=FALSE,message=FALSE}
# identify sample with at least 2 DTAI mutationss
multi_DTAI<-test_set%>%filter(grepl("AML",Dx))%>%
                filter((ASXL1+DNMT3A+TET2+IDH1+IDH2)>=2)%>%
                distinct(Sample)%>%pull(Sample)


# identify gene pairs in the dominant clone
genes_in_each_dominant_clone<-do.call(rbind,lapply(multi_DTAI,function(sample){
  # select the clones
  clones<-final_sample_summary[[sample]]$Clones%>%select(Clone)
  # Extract the mutations
  mutations <- colnames(final_sample_summary[[sample]]$NGT%>%select(!Clone))
  # Identify the dominant clone, which is the last one in this data frame
  WT_clone_to_exclude <-!apply(do.call(rbind,strsplit(as.character(clones[,1]),split="_")),1,function(x){all(x=="0")})
  dominant_clone <- tail(clones[WT_clone_to_exclude,],n=1)
  
  # Separate clones into mutations, and identify the mutations in the dominant clone
  dominant_variants<-final_sample_summary[[sample]]$Clones%>%
                              select(Clone)%>%
                              separate(col=Clone,into=mutations,sep="_",remove=FALSE)%>%
                              pivot_longer(cols=mutations,names_to="Variant",values_to="Genotype")%>%
                              filter(Clone==`dominant_clone`&Genotype!=0)%>%
                              mutate(Gene=do.call(rbind,strsplit(Variant,"[\\._]"))[,1])%>%
                              filter(Gene%in%c(DTAI_genes))%>%
                              pull(Gene)
  
  # Identify dominant clone size
  dominant_clone_size <-final_sample_summary[[sample]]$Clones%>%
                                      summarise(Clone_size=Count[Clone==dominant_clone]/sum(Count)) 
  
  # Create a combinatorial edge list of all mutants in the dominant clone
  if(length(dominant_variants)>=2){
      return(setNames(data.frame(t(combn(dominant_variants,2)),dominant_clone_size,"Dominant"),c("to","from","size","Clonality")))} 
  else if(length(dominant_variants)==1){
      return(setNames(data.frame(t(c(dominant_variants,dominant_variants)),dominant_clone_size,"Dominant"),c("to","from","size","Clonality")))} 
  else if(length(dominant_variants)==0){
      NULL
  }
}))
  


genes_in_each_subclone<-do.call(rbind,lapply(multi_DTAI,function(sample){
   # select the clones
  clones<-final_sample_summary[[sample]]$Clones%>%select(Clone)
  # Extract the mutations
  mutations <- colnames(final_sample_summary[[sample]]$NGT%>%select(!Clone))
  # Identify the dominant clone, which is the last one in this data frame
  WT_clone_to_exclude <-!apply(do.call(rbind,strsplit(as.character(clones[,1]),split="_")),1,function(x){all(x=="0")})
  dominant_clone <- tail(clones[WT_clone_to_exclude,],n=1)
  
  # Separate clones into mutations, and identify the mutations in the dominant clone
  multi_mutant_subclones<-final_sample_summary[[sample]]$Clones%>%
                              mutate(Clone_size=Count/sum(Count))%>%
                              separate(col=Clone,into=mutations,sep="_",remove=FALSE)%>%
                              pivot_longer(cols=mutations,names_to="Variant",values_to="Genotype")%>%
                              filter(Clone!=`dominant_clone`&Genotype!=0)%>%
                              mutate(Gene=do.call(rbind,strsplit(Variant,"[\\._]"))[,1])%>%
                              filter(Gene%in%c(DTAI_genes))%>%
                              group_by(Clone)%>%
                              add_count(Clone)%>%
                              ungroup()%>%
                              filter(n>1)%>%filter(Clone_size==max(Clone_size))
  subclone_variants <- multi_mutant_subclones%>%pull(Gene)
  subclone_size  <- multi_mutant_subclones%>%pull(Clone_size)
    # Create a combinatorial edge list of all mutants in the largest subclone clone
  if(length(subclone_variants)>=2){
      return(setNames(data.frame(t(combn(subclone_variants,2)),subclone_size,"Subclone"),c("to","from","size","Clonality")))} 
  else if(length(subclone_variants)==1){
      return(setNames(data.frame(t(c(subclone_variants,subclone_variants)),subclone_size,"Subclone"),c("to","from","size","Clonality")))} 
  else if(length(subclone_variants)==0){
      NULL
  }
  
}))
  

final_set <- rbind(genes_in_each_dominant_clone,genes_in_each_subclone)%>%
                                        distinct(to,from,size,Clonality)%>%
  # Only doing this next part to match the figure im the paper as best as pssible.
  # After I recoded, the order changed and the changed the order of the graph (though not the overall structure)
                                        arrange(factor(Clonality,levels=c("Dominant","Subclone")),
                                                factor(to, 
                                                       levels = c("IDH2","TET2","ASXL1",
                                                                  "DNMT3A","IDH1" )),
                                                size
                                                )
library(igraph)
set.seed(68864)
graph<-graph_from_data_frame(final_set%>%filter(to!=from),directed=F)%>%
                    set_edge_attr("weight", value = as.numeric(final_set%>%filter(to!=from)%>%pull(size))*3) %>%
                    set_edge_attr("color", value =ifelse(final_set%>%filter(to!=from)%>%pull(Clonality)=="Dominant",  
                                                         brewer.pal(5,"Reds")[5],"grey20"))


mutant_counts<-table(c(as.character(final_set$to),as.character(final_set$from)))[names(V(graph))]
scaled_mutant_counts <-mutant_counts/sum(mutant_counts)*50

radian.rescale <- function(x, start=0, direction=1) {
  c.rotate <- function(x) (x + start) %% (2 * pi) * direction
  c.rotate(scales::rescale(x, c(0, 2 * pi), range(x)))
}

lab.locs <- radian.rescale(x=1:5, direction=-1, start=5)
lab.locs[3]<- -2.5

plot.igraph(graph,edge.width = E(graph)$weight, vertex.color=brewer.pal(5,"Reds")[5],
            vertex.frame.color=brewer.pal(5,"Reds")[5],
            vertex.size=scaled_mutant_counts, vertex.label.family="Helvetica",
            vertex.label.color="black",vertex.label.degree=lab.locs,vertex.label.dist=c(3,4,3,7,3),
            layout=layout_in_circle)


```


```{r, warning=FALSE,message=FALSE}

# identify gene pairs in the dominant clone
genes_in_each_dominant_clone<-do.call(rbind,lapply(multi_DTAI,function(sample){
  # select the clones
  clones<-final_sample_summary[[sample]]$Clones%>%select(Clone)
  # Extract the mutations
  mutations <- colnames(final_sample_summary[[sample]]$NGT%>%select(!Clone))
  # Identify the dominant clone, which is the last one in this data frame
  WT_clone_to_exclude <-!apply(do.call(rbind,strsplit(as.character(clones[,1]),split="_")),1,function(x){all(x=="0")})
  dominant_clone <- tail(clones[WT_clone_to_exclude,],n=1)
  
  # Separate clones into mutations, and identify the mutations in the dominant clone
  dominant_variants<-final_sample_summary[[sample]]$Clones%>%
                              select(Clone)%>%
                              separate(col=Clone,into=mutations,sep="_",remove=FALSE)%>%
                              pivot_longer(cols=mutations,names_to="Variant",values_to="Genotype")%>%
                              filter(Clone==`dominant_clone`&Genotype!=0)%>%
                              mutate(Gene=do.call(rbind,strsplit(Variant,"[\\._]"))[,1])%>%
                              filter(Gene%in%c(signaling_genes))%>%
                              pull(Gene)
  
  # Identify dominant clone size
  dominant_clone_size <-final_sample_summary[[sample]]$Clones%>%
                                      summarise(Clone_size=Count[Clone==dominant_clone]/sum(Count)) 
  
  # Create a combinatorial edge list of all mutants in the dominant clone
  if(length(dominant_variants)>=2){
      return(setNames(data.frame(t(combn(dominant_variants,2)),dominant_clone_size,"Dominant"),c("to","from","size","Clonality")))} 
  else if(length(dominant_variants)==1){
      return(setNames(data.frame(t(c(dominant_variants,dominant_variants)),0,"Dominant"),c("to","from","size","Clonality")))} 
  else if(length(dominant_variants)==0){
      NULL
  }
}))
  


genes_in_each_subclone<-do.call(rbind,lapply(multi_DTAI,function(sample){
   # select the clones
  clones<-final_sample_summary[[sample]]$Clones%>%select(Clone)
  # Extract the mutations
  mutations <- colnames(final_sample_summary[[sample]]$NGT%>%select(!Clone))
  # Identify the dominant clone, which is the last one in this data frame
  WT_clone_to_exclude <-!apply(do.call(rbind,strsplit(as.character(clones[,1]),split="_")),1,function(x){all(x=="0")})
  dominant_clone <- tail(clones[WT_clone_to_exclude,],n=1)
  
  # Separate clones into mutations, and identify the mutations in the dominant clone
  multi_mutant_subclones<-final_sample_summary[[sample]]$Clones%>%
                              mutate(Clone_size=Count/sum(Count))%>%
                              separate(col=Clone,into=mutations,sep="_",remove=FALSE)%>%
                              pivot_longer(cols=mutations,names_to="Variant",values_to="Genotype")%>%
                              filter(Clone!=`dominant_clone`&Genotype!=0)%>%
                              mutate(Gene=do.call(rbind,strsplit(Variant,"[\\._]"))[,1])%>%
                              filter(Gene%in%c(signaling_genes))%>%
                              group_by(Clone)%>%
                              add_count(Clone)%>%
                              ungroup()%>%
                              filter(n>1)%>%
                              filter(Clone_size==max(Clone_size))
  subclone_variants <- multi_mutant_subclones%>%pull(Gene)
  subclone_size  <- multi_mutant_subclones%>%pull(Clone_size)
    # Create a combinatorial edge list of all mutants in the largest subclone clone
  if(length(subclone_variants)>=2){
      return(setNames(data.frame(t(combn(subclone_variants,2)),subclone_size,"Subclone"),c("to","from","size","Clonality")))} 
  else if(length(subclone_variants)==1){
      return(setNames(data.frame(t(c(subclone_variants,subclone_variants)),0,"Subclone"),c("to","from","size","Clonality")))} 
  else if(length(subclone_variants)==0){
      NULL
  }
  
}))
  
final_set <- rbind(genes_in_each_dominant_clone,genes_in_each_subclone)%>%
                                        distinct(to,from,size,Clonality)%>%
  # Only doing this next part to match the figure im the paper as best as pssible.
  # After I recoded, the order changed and the changed the order of the graph (though not the overall structure)
                                         arrange(factor(Clonality,levels=c("Dominant","Subclone")),
                                                factor(to, 
                                                 levels = c("KRAS","JAK2","PTPN11",
                                                                  "FLT3","NRAS" )),
                                                size
                                                ) 

graph<-graph_from_data_frame(final_set%>%filter(to!=from),directed=F)%>%
                    set_edge_attr("weight", value = as.numeric(final_set%>%filter(to!=from)%>%pull(size))*3) %>%
                    set_edge_attr("color", value =ifelse(final_set%>%filter(to!=from)%>%pull(Clonality)=="Dominant",  
                                                         brewer.pal(5,"Reds")[5],"grey20"))


mutant_counts<-table(c(as.character(final_set$to),as.character(final_set$from)))[names(V(graph))]
scaled_mutant_counts <-mutant_counts/sum(mutant_counts)*50


radian.rescale <- function(x, start=0, direction=1) {
  c.rotate <- function(x) (x + start) %% (2 * pi) * direction
  c.rotate(scales::rescale(x, c(0, 2 * pi), range(x)))
}

lab.locs <- radian.rescale(x=1:5, direction=-1, start=5)
lab.locs[3]<- -2.5

plot.igraph(graph,edge.width = E(graph)$weight, vertex.color=brewer.pal(5,"Reds")[5],
            vertex.frame.color=brewer.pal(5,"Reds")[5],
            vertex.size=scaled_mutant_counts, vertex.label.family="Helvetica",
            vertex.label.color="black",vertex.label.degree=lab.locs,vertex.label.dist=c(3,4,3,7,3),
            layout=layout_in_circle)


```

### Clone level co-mutations
```{r eval=FALSE}

# We are going to retabulate what we had before but now on a per clone levels
mutants_in_each_clone<-do.call(rbind,lapply(names(final_sample_summary),function(sample){
  
# select the clones
  clones<-final_sample_summary[[sample]]$Clones%>%select(Clone)
  # Extract the mutations
  mutations <- colnames(final_sample_summary[[sample]]$NGT%>%select(!Clone))

  # Separate clones into mutations, and identify the mutations in the dominant clone
  out<-final_sample_summary[[sample]]$Clones%>%
                              mutate(Clone_size=Count/sum(Count))%>%
                              select(Clone,Clone_size)%>%
                              separate(col=Clone,into=mutations,sep="_",remove=FALSE)%>%
                              pivot_longer(cols=mutations,names_to="Variant",values_to="Genotype")%>%
                              add_column(Sample=`sample`)%>%
                              group_by(Clone)%>%
                              mutate(WT=ifelse(all(Genotype==0),1,0))%>%
                              filter(WT==0)%>%
                              filter(Genotype!=0)%>%
                              ungroup()%>%
                              mutate(Clonality=ifelse(Clone_size==max(Clone_size),
                                                      "Dominant","Subclone"))%>%
                              group_by(Clone)%>%
                              mutate(Gene=do.call(rbind,strsplit(Variant,"[\\._]"))[,1])%>%
                              mutate(DNMT3A=ifelse(any(Gene%in%"DNMT3A"),1,0),
                                     TET2=ifelse(any(Gene%in%"TET2"),1,0),
                                     ASXL1=ifelse(any(Gene%in%"ASXL1"),1,0),
                                     IDH1=ifelse(any(Gene%in%"IDH1"),1,0),
                                     IDH2=ifelse(any(Gene%in%"IDH2"),1,0),
                                     FLT3=ifelse(any(Gene%in%"FLT3"),1,0),
                                     NRAS=ifelse(any(Gene%in%"NRAS"),1,0),
                                     KRAS=ifelse(any(Gene%in%"KRAS"),1,0),
                                     PTPN11=ifelse(any(Gene%in%"PTPN11"),1,0),
                                     JAK2=ifelse(any(Gene%in%"JAK2"),1,0))%>%
                              ungroup()%>%
                              select(!c(Variant,Genotype,Gene))%>%
                              distinct()
  }))


mutants_in_each_sample$Group <- ifelse(grepl("CH",mutants_in_each_sample$Sample),"CH","AML")
mutants_in_each_sample[mutants_in_each_sample$Sample%in%c("MA1715B","MA6300A","MA9521A","MA9521B"),"Group"]<- "MPN"
mutants_in_each_sample[mutants_in_each_sample$Sample%in%c("E4840","E4838new"),"Group"]<- "CMML"
mutants_in_each_sample[mutants_in_each_sample$Sample%in%c("MA4244A","MA2725","MA9521B","MA0092A","R2715","MA1715B","MA6363B"),"Group"]<- "MF"
temp_mat <- as.matrix(mutants_in_each_sample[,6:10])
mode(temp_mat) <- "numeric"
mutants_in_each_sample$signal2 <- ifelse(rowSums(temp_mat)>=2,1,0)


patient_samples <- list(
  "DNMT3A IDH1"=mutants_in_each_sample%>%filter(Group=="AML")%>%filter(TET2==0&DNMT3A==1&IDH1==1&IDH2==0&ASXL1==0)%>%pull(Sample),
  "DNMT3A IDH2"=mutants_in_each_sample%>%filter(Group=="AML")%>%filter(TET2==0&DNMT3A==1&IDH1==0&IDH2==1&ASXL1==0)%>%pull(Sample),
  "DNMT3A_samples"=mutants_in_each_sample%>%filter(Group=="AML")%>%filter(TET2==0&DNMT3A==1&IDH1==0&IDH2==0&ASXL1==0)%>%pull(Sample),
  "IDH1_samples"=mutants_in_each_sample%>%filter(Group=="AML")%>%filter(TET2==0&DNMT3A==0&IDH1==1&IDH2==0&ASXL1==0)%>%pull(Sample),
  "IDH2_samples"=mutants_in_each_sample%>%filter(Group=="AML")%>%filter(TET2==0&DNMT3A==0&IDH1==0&IDH2==1&ASXL1==0)%>%pull(Sample))


comutant_status<-do.call(rbind,lapply(names(patient_samples), function(gene_pair){
  x <-unique(patient_samples[[gene_pair]])
  data.frame("Total"=mutants_in_each_sample%>%filter(Sample%in%x)%>%summarise(Count=n())%>%pull(Count),
             "FLT3"=mutants_in_each_sample%>%filter(Sample%in%x&signal2==0)%>%tally(FLT3)%>%pull(n),
            "PTPN11"=mutants_in_each_sample%>%filter(Sample%in%x&signal2==0)%>%tally(PTPN11)%>%pull(n),
             "JAK2"=mutants_in_each_sample%>%filter(Sample%in%x&signal2==0)%>%tally(JAK2)%>%pull(n),
          "KRAS"=mutants_in_each_sample%>%filter(Sample%in%x&signal2==0)%>%tally(KRAS)%>%pull(n),
         # "IDH1"=mutants_in_each_sample%>%filter(Sample%in%x&signal2==0)%>%tally(IDH1)%>%pull(n),
        #  "IDH2"=mutants_in_each_sample%>%filter(Sample%in%x&signal2==0)%>%tally(IDH2)%>%pull(n),
          "NRAS"=mutants_in_each_sample%>%filter(Sample%in%x&signal2==0)%>%tally(NRAS)%>%pull(n),
             "Multiple mutants"=mutants_in_each_sample%>%filter(Sample%in%x)%>%tally(signal2)%>%pull(n))
}))
  
rownames(comutant_status)<-  names(patient_samples)
  
comutant_status$None <-comutant_status[,"Total"]-rowSums(comutant_status[,-1])
comutant_status$Group <- rownames(comutant_status)
final<-comutant_status[,-1]%>%melt()
final$Group <- factor(do.call(rbind,strsplit(final$Group,split="_"))[,1],levels=rev(c("IDH1","DNMT3A IDH1","DNMT3A","DNMT3A IDH2","IDH2")))
final$variable <- gsub("\\."," ",final$variable)
final$variable <- factor(final$variable,levels=rev(c("JAK2","PTPN11","NRAS","KRAS","FLT3","Multiple mutants","None")))

color_set<-rev(brewer.pal(9,"Set1")[c(1,8,2,3,4,5,9)])

gg_co_mutants<-ggplot(final, aes(x=Group,fill=variable,y=value))+geom_bar(stat="identity",position="fill")+
                theme_classic(base_size=8)+
                ylab("Fraction of clones")+
                xlab("")+
                coord_flip()+
               # theme(axis.text.x = element_text(angle=30,hjust=1))+
                scale_y_continuous(expand=c(0,0))+
                scale_fill_manual(values=color_set,"Co mutation",guide = guide_legend(reverse = TRUE))


ggsave(addSmallLegend(gg_co_mutants),
       width=3.5,height=1.5,
       file="/Volumes/LevineLab/Levine Lab/Bobby/Collaborations/MissionBio/Analysis/2020/March/Figures/Figure 2G New.pdf")

multi_mutant <- unlist(patient_samples[1:2])
patients_of_interest<-mutants_in_each_sample%>%filter(signal2==1&Sample%in%multi_mutant)%>%pull(Sample)


output_folder <- "/Users/bowmanr/Desktop/mBio_test/"  

mutually_exclusive<-setNames(lapply(patients_of_interest,function(y){
  x<-(final_sample_summary[[y]]$NGT)
  x<-x[,!grepl("Clone",colnames(x))]
  x[is.na(x)] <-0
  x[x>0] <-1
  x<- as.matrix(x)
  mode(x) <-"numeric"
  grob_corrplot_single_cell_loop<-generate_and_plot_cooccurence(x)
}),patients_of_interest)

lapply(patients_of_interest,function(sample){
data.frame(mutually_exclusive[[sample]]$data)[,10:12]#%>%filter(score!=0)
})


mutually_exclusive<-setNames(lapply(patients_of_interest,function(y){
  x<-(final_sample_summary[[y]]$NGT)
  x<-x[,!grepl("Clone",colnames(x))]
  x[is.na(x)] <-0
  x[x>0] <-1
  epi<-x[,grepl("DNMT3A",colnames(x))|grepl("TET2",colnames(x))|grepl("IDH1",colnames(x))|grepl("IDH2",colnames(x))]
  signal<-x[,grepl("PTPN11",colnames(x))|grepl("JAK2",colnames(x))|grepl("FLT3",colnames(x))|grepl("NRAS",colnames(x))|grepl("KRAS",colnames(x))]
  
  data.frame("Epigenetic"=sum(apply(epi,1,function(z){sum(z==1)>=2}))/nrow(epi),
             "Signaling"=sum(apply(signal,1,function(z){sum(z==1)>=2}))/nrow(signal))
         
}),patients_of_interest)

gg_fraction_comutated_cells<-ggplot(melt(do.call(rbind,mutually_exclusive)),aes(x=variable,y=value,fill=variable))+
                    geom_boxplot()+
                    geom_jitter(width=0.1)+
                    theme_classic(base_size=8)+
                    scale_fill_brewer(type="qual",palette = "Set1","Mutation pairs")+
                    xlab("")+ylab("Fraction of co-mutated cells")

melt(do.call(rbind,mutually_exclusive))%>%summarise(p=t.test(value~variable)$p.value)


ggsave((gg_fraction_comutated_cells),
       width=2.5,height=2,
       file="/Volumes/LevineLab/Levine Lab/Bobby/Collaborations/MissionBio/Analysis/2020/January/Figures/EF3F-comutant_cell_fraction.pdf")




```