<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>5.1 Import from Tapestri Insights | Single cell mutational profiling delineates clonal trajectories in myeloid malignancies</title>
  <meta name="description" content="This is a code base for all of the analyses performed in Miles, Bowman et al." />
  <meta name="generator" content="bookdown 0.18.3 and GitBook 2.6.7" />

  <meta property="og:title" content="5.1 Import from Tapestri Insights | Single cell mutational profiling delineates clonal trajectories in myeloid malignancies" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This is a code base for all of the analyses performed in Miles, Bowman et al." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="5.1 Import from Tapestri Insights | Single cell mutational profiling delineates clonal trajectories in myeloid malignancies" />
  
  <meta name="twitter:description" content="This is a code base for all of the analyses performed in Miles, Bowman et al." />
  

<meta name="author" content="Robert Bowman" />


<meta name="date" content="2020-05-17" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="Appendix.html"/>
<link rel="next" href="functions.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/kePrint-0.0.1/kePrint.js"></script>


<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Minimal Book Example</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Foreward</a></li>
<li class="chapter" data-level="2" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i><b>2</b> Introduction</a><ul>
<li class="chapter" data-level="2.1" data-path="data-extraction-and-setup.html"><a href="data-extraction-and-setup.html"><i class="fa fa-check"></i><b>2.1</b> Data Extraction and Setup</a></li>
<li class="chapter" data-level="2.2" data-path="tapestri-library.html"><a href="tapestri-library.html"><i class="fa fa-check"></i><b>2.2</b> Tapestri Package</a></li>
<li class="chapter" data-level="2.3" data-path="post-processing.html"><a href="post-processing.html"><i class="fa fa-check"></i><b>2.3</b> Post processing</a></li>
<li class="chapter" data-level="2.4" data-path="assessing-clonal-abundance.html"><a href="assessing-clonal-abundance.html"><i class="fa fa-check"></i><b>2.4</b> Assessing clonal abundance</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="manuscript-analysis.html"><a href="manuscript-analysis.html"><i class="fa fa-check"></i><b>3</b> Manuscript analysis</a><ul>
<li class="chapter" data-level="3.1" data-path="figure-1-cohort-characterization.html"><a href="figure-1-cohort-characterization.html"><i class="fa fa-check"></i><b>3.1</b> Figure 1: Cohort characterization</a><ul>
<li class="chapter" data-level="3.1.1" data-path="figure-1-cohort-characterization.html"><a href="figure-1-cohort-characterization.html#cohort-characteristics-ef1a-d"><i class="fa fa-check"></i><b>3.1.1</b> Cohort characteristics (EF1A-D)</a></li>
<li class="chapter" data-level="3.1.2" data-path="figure-1-cohort-characterization.html"><a href="figure-1-cohort-characterization.html#mutation-co-occurence"><i class="fa fa-check"></i><b>3.1.2</b> Mutation Co-occurence</a></li>
<li class="chapter" data-level="3.1.3" data-path="figure-1-cohort-characterization.html"><a href="figure-1-cohort-characterization.html#oncoprint-figure-1a"><i class="fa fa-check"></i><b>3.1.3</b> Oncoprint: Figure 1A</a></li>
<li class="chapter" data-level="3.1.4" data-path="figure-1-cohort-characterization.html"><a href="figure-1-cohort-characterization.html#sample-clonality-figure-1ce-figure-2ab"><i class="fa fa-check"></i><b>3.1.4</b> Sample clonality: Figure 1C,E Figure 2A,B</a></li>
<li class="chapter" data-level="3.1.5" data-path="figure-1-cohort-characterization.html"><a href="figure-1-cohort-characterization.html#clonograph-figure-1d"><i class="fa fa-check"></i><b>3.1.5</b> Clonograph: Figure 1D</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="figure-2-clonality.html"><a href="figure-2-clonality.html"><i class="fa fa-check"></i><b>3.2</b> Figure 2: Clonality</a><ul>
<li class="chapter" data-level="3.2.1" data-path="figure-2-clonality.html"><a href="figure-2-clonality.html#clonality-in-disease-states"><i class="fa fa-check"></i><b>3.2.1</b> Clonality in disease states</a></li>
<li class="chapter" data-level="3.2.2" data-path="figure-2-clonality.html"><a href="figure-2-clonality.html#clonality-in-co-mutational-groups"><i class="fa fa-check"></i><b>3.2.2</b> Clonality in co-mutational groups</a></li>
<li class="chapter" data-level="3.2.3" data-path="figure-2-clonality.html"><a href="figure-2-clonality.html#clonality-assocation-with-vaf"><i class="fa fa-check"></i><b>3.2.3</b> Clonality assocation with VAF</a></li>
<li class="chapter" data-level="3.2.4" data-path="figure-2-clonality.html"><a href="figure-2-clonality.html#co-mutation-and-clonality"><i class="fa fa-check"></i><b>3.2.4</b> Co-mutation and clonality</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="final-words.html"><a href="final-words.html"><i class="fa fa-check"></i><b>4</b> Final Words</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="chapter" data-level="5" data-path="Appendix.html"><a href="Appendix.html"><i class="fa fa-check"></i><b>5</b> Appendix</a><ul>
<li class="chapter" data-level="5.1" data-path="AppendixA.html"><a href="AppendixA.html"><i class="fa fa-check"></i><b>5.1</b> Import from Tapestri Insights</a></li>
<li class="chapter" data-level="5.2" data-path="functions.html"><a href="functions.html"><i class="fa fa-check"></i><b>5.2</b> Functions</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Single cell mutational profiling delineates clonal trajectories in myeloid malignancies</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="AppendixA" class="section level2">
<h2><span class="header-section-number">5.1</span> Import from Tapestri Insights</h2>
<p><em>Here is how we used to do it and what I wrote last time:</em> Our first step is to extract the data from output from Tapestri Insights. To do so, I’ve written a few functions that will be helpful to do this on a long list of samples. As this project grew it becamse clear that this work was better suited for a package. Before that though, looking forward we’ll plan to analyze data using the tapestri R package and integrate formatting from standard single cell rna-seq data in R. For now, we’ll just source a series of functions that can be found in the appendix. I’ll build in explanations as best I can over time.</p>
<p>Below I’ve copied directly the README.txt file output from Tapestri Insights, as it most accurately describes the output from each of the files.</p>
<p><span style="color: blue;"> This folder contains data exported from Tapestri Insights 2.1 application. The following files are included:<br />
<strong>DP.csv</strong> - read depth. The Read Depth per Variant metric is the filtered depth, at the cell level. This gives the number of filtered reads that support each of the reported alleles.<br />
<strong>GQ.csv</strong> - quality scores. The genotype quality score represents the Phred-scaled confidence that the genotype assignment (GT) is correct, derived from the genotype normalized likelihoods of possible genotypes (PL). Specifically, the quality score is the difference between the PL of the second most likely genotype, and the PL of the most likely genotype. The values of the PLs are normalized so that the most likely PL is always 0, so the quality score ends up being equal to the second smallest PL, unless that PL is greater than 99. In GATK, the value of quality score is capped at 99 because larger values are not more informative, but they take more space in the file. So if the second most likely PL is greater than 99, we still assign a quality score of 99. Basically the GQ gives you the difference between the likelihoods of the two most likely genotypes. If it is low, you can tell there is not much confidence in the genotype, i.e. there was not enough evidence to confidently choose one genotype over another.<br />
<strong>NGT.csv</strong> - genotype call converted to categorical (0-reference, 1-heterozygous mutation, 2-homozygous mutation, 3-unknown). Calls are made by GATK/Haplotypecaller.<br />
<strong>AF.csv</strong> - variant allele frequency (# of reads with evidence for mutation / total # of reads * 100)<br />
<strong>Variants.csv</strong> - variant metadata</span></p>
<blockquote>
<p><strong>In the weeds a little here to explain a detour</strong><br />
It is worth mentioning here on why we chose to proceed in the following fashion, and why we didn’t do all the parsing in R directly from the loom file. When we first started running these samples, there were some incompatibilities with the loom v3 files produced from Bluebee, and the available packages in R that were only working with loom v2. This has now of course been corrected. We considered using python, but the reality was my proficiency in R was significantly greater than python and this was not a tennable solutuon. Going through Tapestri Insights also had two added benefits: 1) efficient SNV to protein identification, and 2) data access for others in the lab who were not quickly pouring into the data in R. Future projects in our lab are likely to bypass Tapestri Insights, and bring loom files straight into R as you’ll see in the DNA+Protein section. Furthermore, as this project progressed, new versions of Tapestri Insights became available, and this led to minor modifications in column names and file names. All of this was parsed in R, and the version of Tapestri Insight did not change the analysis, we just did not rerun &gt;100 samples through Tapestri Insights to make sure it was all formatted directly. This is all to explain why you will see some patchwork code in some of the extraction functions upcoming in the next section.</p>
</blockquote>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">all_samples &lt;-<span class="st"> </span><span class="kw">list.files</span>(<span class="st">&quot;/Volumes/LevineLab/Levine Lab/MissionBio_Tapestri/Insights_Output/ASH_Cohort&quot;</span>)
blacklist &lt;-<span class="st"> </span><span class="kw">read.xls</span>(<span class="st">&quot;/Volumes/LevineLab/Levine Lab/Bobby/Collaborations/MissionBio/From_LAM/Blacklist2.xlsx&quot;</span>)
whitelist &lt;-<span class="st"> </span><span class="kw">read.xls</span>(<span class="st">&quot;/Volumes/LevineLab/Levine Lab/Bobby/Collaborations/MissionBio/From_LAM/Whitelist.xlsx&quot;</span>)
pheno &lt;-<span class="st"> </span><span class="kw">read.xls</span>(<span class="st">&quot;/Volumes/LevineLab/Levine Lab/Bobby/Collaborations/MissionBio/From_LAM/pheno_MBio.xlsx&quot;</span>)
sample_set&lt;-<span class="st"> </span><span class="kw">intersect</span>(<span class="kw">as.character</span>(all_samples),<span class="kw">as.character</span>(pheno<span class="op">$</span>Sample.ID)) 

<span class="kw">setwd</span>(<span class="st">&quot;/Volumes/LevineLab/Levine Lab/MissionBio_Tapestri/Insights_Output/ASH_Cohort&quot;</span>)
sample_SNPS&lt;-<span class="kw">lapply</span>(<span class="kw">as.list</span>(sample_set),<span class="cf">function</span>(x){
  y&lt;-<span class="kw">extract_SNP_info</span>(x)
  z &lt;-<span class="st"> </span>y[<span class="op">!</span>(<span class="kw">rownames</span>(y)<span class="op">%in%</span>blacklist[,<span class="st">&quot;Blacklisted.coordinate&quot;</span>]<span class="op">|</span>y[,<span class="st">&quot;protein&quot;</span>]<span class="op">%in%</span>blacklist[,<span class="st">&quot;Protein.Change&quot;</span>]),]
  z[,<span class="st">&quot;SNP_variant&quot;</span>] &lt;-<span class="st"> </span><span class="kw">rownames</span>(z)
  z[,<span class="st">&quot;Sample&quot;</span>] &lt;-<span class="st"> </span>x
  <span class="kw">return</span>(z)
})

<span class="kw">names</span>(sample_SNPS) &lt;-<span class="st"> </span>sample_set
<span class="kw">saveRDS</span>(sample_SNPS,<span class="dt">file=</span><span class="st">&quot;/Volumes/LevineLab/Levine Lab/Bobby/Collaborations/MissionBio/Analysis/2020/January/sample_SNPS.rds&quot;</span>)

sample_NGTs&lt;-<span class="kw">lapply</span>(<span class="kw">as.list</span>(<span class="kw">names</span>(sample_SNPS)),<span class="cf">function</span>(x){
  <span class="kw">extract_NGT_files</span>(x,sample_SNPS[[x]]<span class="op">$</span>SNP_variant)
})
<span class="kw">names</span>(sample_NGTs) &lt;-<span class="st"> </span><span class="kw">names</span>(sample_SNPS)

named_sample_NGTs&lt;-<span class="kw">lapply</span>(sample_NGTs,<span class="cf">function</span>(x){
  <span class="kw">rownames</span>(x) &lt;-<span class="st"> </span><span class="kw">paste</span>(<span class="st">&quot;Cell&quot;</span>,<span class="dv">1</span><span class="op">:</span><span class="kw">nrow</span>(x),<span class="dt">sep=</span><span class="st">&quot;_&quot;</span>)
  <span class="kw">return</span>(x)
})
<span class="kw">saveRDS</span>(named_sample_NGTs,<span class="dt">file=</span><span class="st">&quot;/Volumes/LevineLab/Levine Lab/Bobby/Collaborations/MissionBio/Analysis/2020/January/sample_NGTs.rds&quot;</span>)


pheno &lt;-<span class="st"> </span><span class="kw">read.xls</span>(<span class="st">&quot;/Volumes/LevineLab/Levine Lab/Bobby/Collaborations/MissionBio/From_LAM/pheno_MBio.xlsx&quot;</span>)

<span class="co"># Now we filter out any synonymous mutations as well</span>
sample_SNPS_filter&lt;-<span class="kw">lapply</span>(sample_SNPS,<span class="cf">function</span>(x){ x<span class="op">%&gt;%</span><span class="kw">filter</span>(<span class="op">!</span>(<span class="kw">grepl</span>(<span class="st">&quot;[=&gt;]&quot;</span>,protein)))<span class="op">%&gt;%</span><span class="kw">filter</span>(<span class="op">!</span><span class="kw">grepl</span>(<span class="st">&quot;ASXL1:p.[DNGPIL]81&quot;</span>,protein))<span class="op">%&gt;%</span><span class="kw">filter</span>(<span class="op">!</span><span class="kw">grepl</span>(<span class="st">&quot;FLT3_INS_chr13:28602226:&quot;</span>,protein)) }) 
colnames_to_grab &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Variant&quot;</span>,<span class="st">&quot;protein&quot;</span>,<span class="st">&quot;Gene&quot;</span>,<span class="st">&quot;gene&quot;</span>,<span class="st">&quot;allele_freq_loom&quot;</span>,<span class="st">&quot;allele_freq&quot;</span>,<span class="st">&quot;SNP_variant&quot;</span>,<span class="st">&quot;Sample&quot;</span>)
sample_SNPS_filter_cols&lt;-<span class="kw">lapply</span>(sample_SNPS_filter,<span class="cf">function</span>(x){x[,<span class="kw">colnames</span>(x)<span class="op">%in%</span>colnames_to_grab]}) 

sample_NGTs_filter&lt;-<span class="kw">lapply</span>(<span class="kw">names</span>(sample_SNPS),<span class="cf">function</span>(x){<span class="kw">setNames</span>(<span class="kw">data.frame</span>(sample_NGTs[[x]][,sample_SNPS_filter[[x]][,<span class="st">&quot;SNP_variant&quot;</span>]]),sample_SNPS_filter[[x]][,<span class="st">&quot;SNP_variant&quot;</span>])}); <span class="kw">names</span>(sample_NGTs_filter) &lt;-<span class="st"> </span><span class="kw">names</span>(sample_SNPS)
final_variants_of_interest&lt;-<span class="kw">lapply</span>(sample_NGTs_filter,<span class="cf">function</span>(x){<span class="kw">colnames</span>(x)[<span class="kw">colSums</span>(x)<span class="op">&gt;</span><span class="dv">0</span>]})
<span class="kw">names</span>(final_variants_of_interest) &lt;-<span class="kw">names</span>(sample_SNPS)

test_NGT &lt;-<span class="st"> </span><span class="kw">setNames</span>(<span class="kw">lapply</span>(<span class="kw">names</span>(final_variants_of_interest),<span class="cf">function</span>(x){<span class="kw">setNames</span>(<span class="kw">data.frame</span>(sample_NGTs_filter[[x]][,final_variants_of_interest[[x]]]),final_variants_of_interest[[x]] )}),<span class="kw">names</span>(sample_SNPS))

final_NGT&lt;-<span class="kw">setNames</span>(<span class="kw">lapply</span>(<span class="kw">names</span>(test_NGT),<span class="cf">function</span>(x){
  <span class="cf">if</span>(<span class="kw">ncol</span>(test_NGT[[x]])<span class="op">&gt;</span><span class="dv">1</span>){
    j&lt;-test_NGT[[x]][,<span class="kw">apply</span>(test_NGT[[x]],<span class="dv">2</span>,<span class="cf">function</span>(z){<span class="kw">sum</span>(z<span class="op">%in%</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))<span class="op">&gt;=</span><span class="dv">2</span>})]
    z&lt;-j[<span class="op">!</span><span class="kw">apply</span>(j,<span class="dv">1</span>,<span class="cf">function</span>(y){<span class="kw">any</span>(y<span class="op">==</span><span class="dv">3</span>)}),]
    q&lt;-z[,<span class="kw">apply</span>(z,<span class="dv">2</span>,<span class="cf">function</span>(h){<span class="kw">sum</span>(h<span class="op">%in%</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))<span class="op">&gt;=</span><span class="dv">2</span>})]
  } <span class="cf">else</span>{
    j&lt;-<span class="kw">data.frame</span>(test_NGT[[x]][test_NGT[[x]]<span class="op">==</span><span class="dv">1</span><span class="op">|</span>test_NGT[[x]]<span class="op">==</span><span class="dv">2</span>])
    q&lt;-<span class="kw">data.frame</span>(j[<span class="op">!</span>j<span class="op">==</span><span class="dv">3</span>])      
  }
  <span class="kw">return</span>(q)
}),<span class="kw">names</span>(test_NGT))

final_variants_of_interest &lt;-<span class="st"> </span><span class="kw">lapply</span>(final_NGT,colnames)

final_SNP &lt;-<span class="kw">setNames</span>(<span class="kw">lapply</span>(<span class="kw">names</span>(final_variants_of_interest),<span class="cf">function</span>(x){
  sample_SNPS_filter_cols[[x]]<span class="op">%&gt;%</span><span class="kw">filter</span>(SNP_variant<span class="op">%in%</span>final_variants_of_interest[[x]])}),<span class="kw">names</span>(sample_SNPS))

final_SNP_mat&lt;-<span class="kw">do.call</span>(bind_rows,final_SNP)
final_SNP_mat<span class="op">$</span>Gene &lt;-<span class="st"> </span><span class="kw">ifelse</span>(final_SNP_mat<span class="op">$</span>Gene<span class="op">==</span><span class="st">&quot;&quot;</span>,<span class="st">&quot;FLT3&quot;</span>,final_SNP_mat<span class="op">$</span>Gene)
final_SNP_mat<span class="op">$</span>gene &lt;-<span class="st"> </span><span class="kw">ifelse</span>(final_SNP_mat<span class="op">$</span>gene<span class="op">==</span><span class="st">&quot;&quot;</span>,<span class="st">&quot;FLT3&quot;</span>,final_SNP_mat<span class="op">$</span>gene)
final_SNP_mat<span class="op">$</span>Gene &lt;-<span class="st"> </span><span class="kw">ifelse</span>(<span class="kw">is.na</span>(final_SNP_mat<span class="op">$</span>Gene),final_SNP_mat<span class="op">$</span>gene,final_SNP_mat<span class="op">$</span>Gene) 
<span class="kw">colnames</span>(final_SNP_mat)[<span class="dv">5</span>] &lt;-<span class="st"> &quot;Sample.ID&quot;</span>

pheno_mut_melted&lt;-<span class="kw">inner_join</span>(pheno,final_SNP_mat[,<span class="dv">2</span><span class="op">:</span><span class="dv">5</span>])
pheno_mut_melted<span class="op">$</span>Gene&lt;-<span class="st"> </span><span class="kw">factor</span>(pheno_mut_melted<span class="op">$</span>Gene,<span class="dt">levels=</span><span class="kw">names</span>(<span class="kw">sort</span>(<span class="kw">table</span>(pheno_mut_melted<span class="op">$</span>Gene), <span class="dt">decreasing=</span><span class="ot">TRUE</span>)))

<span class="cf">for</span>(x <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(<span class="kw">names</span>(final_NGT))){
  <span class="kw">colnames</span>(final_NGT[x][[<span class="dv">1</span>]])&lt;-pheno_mut_melted[pheno_mut_melted<span class="op">$</span>SNP_variant<span class="op">%in%</span><span class="kw">colnames</span>(final_NGT[[x]])<span class="op">&amp;</span>
<span class="st">                                                  </span>pheno_mut_melted<span class="op">$</span>Sample.ID<span class="op">==</span><span class="kw">names</span>(final_NGT)[x],<span class="st">&quot;protein&quot;</span>]
  <span class="kw">rownames</span>(final_NGT[x][[<span class="dv">1</span>]])&lt;-<span class="kw">paste</span>(<span class="st">&quot;Cell&quot;</span>,<span class="dv">1</span><span class="op">:</span><span class="kw">dim</span>(final_NGT[x][[<span class="dv">1</span>]])[[<span class="dv">1</span>]],<span class="dt">sep=</span><span class="st">&quot;_&quot;</span>)
}


<span class="co">#These lines were used to regenerated a new blacklist</span>
protein_variant_counts&lt;-<span class="kw">data.frame</span>(<span class="kw">sort</span>(<span class="kw">table</span>(pheno_mut_melted[,<span class="st">&quot;protein&quot;</span>]),<span class="dt">decreasing=</span><span class="ot">TRUE</span>))
SNP_variant_counts&lt;-<span class="kw">data.frame</span>(<span class="kw">sort</span>(<span class="kw">table</span>(pheno_mut_melted[,<span class="st">&quot;SNP_variant&quot;</span>]),<span class="dt">decreasing=</span><span class="ot">TRUE</span>))
Muts_per_patientx&lt;-<span class="kw">data.frame</span>(<span class="kw">sort</span>(<span class="kw">table</span>(pheno_mut_melted[,<span class="st">&quot;Sample.ID&quot;</span>]),<span class="dt">decreasing=</span><span class="ot">TRUE</span>))

processing_NGT &lt;-<span class="st"> </span>final_NGT
<span class="kw">names</span>(processing_NGT) &lt;-<span class="st"> </span><span class="kw">names</span>(final_NGT)

cell_number_per_sample_cutoff&lt;-<span class="dv">40</span>
final_sample_set&lt;-<span class="kw">names</span>(<span class="kw">which</span>(<span class="kw">do.call</span>(rbind,<span class="kw">lapply</span>(processing_NGT,dim))[,<span class="dv">1</span>]<span class="op">&gt;</span>cell_number_per_sample_cutoff))

samples_of_interest&lt;-<span class="kw">setdiff</span>(<span class="kw">names</span>(processing_NGT),final_sample_set)
<span class="kw">lapply</span>(processing_NGT[samples_of_interest],nrow)
<span class="kw">lapply</span>(sample_NGTs[samples_of_interest],nrow)

cells_of_interest&lt;-<span class="kw">setNames</span>(<span class="kw">lapply</span>(processing_NGT[final_sample_set],<span class="cf">function</span>(x){
  <span class="kw">rownames</span>(x)
}),final_sample_set)

<span class="kw">saveRDS</span>(processing_NGT[final_sample_set],<span class="dt">file=</span><span class="st">&quot;/Volumes/LevineLab/Levine Lab/Bobby/Collaborations/MissionBio/Analysis/2020/January/final_NGTs.rds&quot;</span>)
<span class="kw">saveRDS</span>(pheno_mut_melted,<span class="dt">file=</span><span class="st">&quot;/Volumes/LevineLab/Levine Lab/Bobby/Collaborations/MissionBio/Analysis/2020/January/pheno_mut_melted.rds&quot;</span>)
<span class="kw">saveRDS</span>(final_sample_set,<span class="dt">file=</span><span class="st">&quot;/Volumes/LevineLab/Levine Lab/Bobby/Collaborations/MissionBio/Analysis/2020/January/final_sample_set.rds&quot;</span>)
<span class="kw">saveRDS</span>(cells_of_interest,<span class="dt">file=</span><span class="st">&quot;/Volumes/LevineLab/Levine Lab/Bobby/Collaborations/MissionBio/Analysis/2020/January/cells_of_interest.rds&quot;</span>)</code></pre></div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="Appendix.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="functions.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["scDNA.pdf", "scDNA.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
