<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>3.1 Figure 1: Cohort characterization | Single cell mutational profiling delineates clonal trajectories in myeloid malignancies</title>
  <meta name="description" content="This is a code base for all of the analyses performed in Miles, Bowman et al. " />
  <meta name="generator" content="bookdown 0.22.2 and GitBook 2.6.7" />

  <meta property="og:title" content="3.1 Figure 1: Cohort characterization | Single cell mutational profiling delineates clonal trajectories in myeloid malignancies" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This is a code base for all of the analyses performed in Miles, Bowman et al. " />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="3.1 Figure 1: Cohort characterization | Single cell mutational profiling delineates clonal trajectories in myeloid malignancies" />
  
  <meta name="twitter:description" content="This is a code base for all of the analyses performed in Miles, Bowman et al. " />
  

<meta name="author" content="Robert Bowman" />


<meta name="date" content="2021-05-10" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="manuscript-analysis.html"/>
<link rel="next" href="figure-2-clonality.html"/>
<script src="libs/header-attrs-2.8/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Minimal Book Example</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Foreward</a></li>
<li class="chapter" data-level="2" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i><b>2</b> Introduction</a>
<ul>
<li class="chapter" data-level="2.1" data-path="data-extraction-and-setup.html"><a href="data-extraction-and-setup.html"><i class="fa fa-check"></i><b>2.1</b> Data Extraction and Setup</a></li>
<li class="chapter" data-level="2.2" data-path="hdf5-and-loom-input.html"><a href="hdf5-and-loom-input.html"><i class="fa fa-check"></i><b>2.2</b> HDF5 and Loom input</a></li>
<li class="chapter" data-level="2.3" data-path="tapestri-package.html"><a href="tapestri-package.html"><i class="fa fa-check"></i><b>2.3</b> Tapestri Package</a></li>
<li class="chapter" data-level="2.4" data-path="post-processing.html"><a href="post-processing.html"><i class="fa fa-check"></i><b>2.4</b> Post processing</a></li>
<li class="chapter" data-level="2.5" data-path="assessing-clonal-abundance.html"><a href="assessing-clonal-abundance.html"><i class="fa fa-check"></i><b>2.5</b> Assessing clonal abundance</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="manuscript-analysis.html"><a href="manuscript-analysis.html"><i class="fa fa-check"></i><b>3</b> Manuscript analysis</a>
<ul>
<li class="chapter" data-level="3.1" data-path="figure-1-cohort-characterization.html"><a href="figure-1-cohort-characterization.html"><i class="fa fa-check"></i><b>3.1</b> Figure 1: Cohort characterization</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="figure-1-cohort-characterization.html"><a href="figure-1-cohort-characterization.html#cohort-characteristics-ef1a-d"><i class="fa fa-check"></i><b>3.1.1</b> Cohort characteristics (EF1A-D)</a></li>
<li class="chapter" data-level="3.1.2" data-path="figure-1-cohort-characterization.html"><a href="figure-1-cohort-characterization.html#mutation-co-occurence"><i class="fa fa-check"></i><b>3.1.2</b> Mutation Co-occurence</a></li>
<li class="chapter" data-level="3.1.3" data-path="figure-1-cohort-characterization.html"><a href="figure-1-cohort-characterization.html#oncoprint-figure-1a"><i class="fa fa-check"></i><b>3.1.3</b> Oncoprint: Figure 1A</a></li>
<li class="chapter" data-level="3.1.4" data-path="figure-1-cohort-characterization.html"><a href="figure-1-cohort-characterization.html#sample-clonality-figure-1ce-figure-2ab"><i class="fa fa-check"></i><b>3.1.4</b> Sample clonality: Figure 1C,E Figure 2A,B</a></li>
<li class="chapter" data-level="3.1.5" data-path="figure-1-cohort-characterization.html"><a href="figure-1-cohort-characterization.html#clonograph-figure-1d"><i class="fa fa-check"></i><b>3.1.5</b> Clonograph: Figure 1D</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="figure-2-clonality.html"><a href="figure-2-clonality.html"><i class="fa fa-check"></i><b>3.2</b> Figure 2: Clonality</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="figure-2-clonality.html"><a href="figure-2-clonality.html#clonality-in-disease-states"><i class="fa fa-check"></i><b>3.2.1</b> Clonality in disease states</a></li>
<li class="chapter" data-level="3.2.2" data-path="figure-2-clonality.html"><a href="figure-2-clonality.html#clonality-in-co-mutational-groups"><i class="fa fa-check"></i><b>3.2.2</b> Clonality in co-mutational groups</a></li>
<li class="chapter" data-level="3.2.3" data-path="figure-2-clonality.html"><a href="figure-2-clonality.html#clonality-assocation-with-vaf"><i class="fa fa-check"></i><b>3.2.3</b> Clonality assocation with VAF</a></li>
<li class="chapter" data-level="3.2.4" data-path="figure-2-clonality.html"><a href="figure-2-clonality.html#co-mutation-and-clonality"><i class="fa fa-check"></i><b>3.2.4</b> Co-mutation and clonality</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="figure-3-genetic-trajectory-analysis.html"><a href="figure-3-genetic-trajectory-analysis.html"><i class="fa fa-check"></i><b>4</b> Figure 3: Genetic Trajectory analysis</a>
<ul>
<li class="chapter" data-level="4.1" data-path="introductory-text-and-philisophy-of-approach.html"><a href="introductory-text-and-philisophy-of-approach.html"><i class="fa fa-check"></i><b>4.1</b> Introductory text and philisophy of approach:</a></li>
<li class="chapter" data-level="4.2" data-path="functions-for-reward-matrix-and-querying-initating-mutations.html"><a href="functions-for-reward-matrix-and-querying-initating-mutations.html"><i class="fa fa-check"></i><b>4.2</b> Functions for reward matrix and querying initating mutations</a></li>
<li class="chapter" data-level="4.3" data-path="executing-the-analysis.html"><a href="executing-the-analysis.html"><i class="fa fa-check"></i><b>4.3</b> Executing the Analysis</a></li>
<li class="chapter" data-level="4.4" data-path="plotting-the-data.html"><a href="plotting-the-data.html"><i class="fa fa-check"></i><b>4.4</b> Plotting the data</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Single cell mutational profiling delineates clonal trajectories in myeloid malignancies</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="figure-1-cohort-characterization" class="section level2" number="3.1">
<h2><span class="header-section-number">3.1</span> Figure 1: Cohort characterization</h2>
<p>Let’s start by loading in some packages that are recurrently used. Other packages that are only used in one chunk of code are listed appropriately below.</p>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="figure-1-cohort-characterization.html#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb35-2"><a href="figure-1-cohort-characterization.html#cb35-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyr)</span>
<span id="cb35-3"><a href="figure-1-cohort-characterization.html#cb35-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb35-4"><a href="figure-1-cohort-characterization.html#cb35-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyselect)</span></code></pre></div>
<p>Here we will focus on the cohort level analysis, and load in the NGT files <em>before</em> they were filtered for clonality.</p>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="figure-1-cohort-characterization.html#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">&quot;/Users/bowmanr/Projects/scDNA&quot;</span>)</span>
<span id="cb36-2"><a href="figure-1-cohort-characterization.html#cb36-2" aria-hidden="true" tabindex="-1"></a>final_NGTs<span class="ot">&lt;-</span><span class="fu">readRDS</span>(<span class="at">file=</span><span class="st">&quot;./data/final_NGTs.rds&quot;</span>)</span>
<span id="cb36-3"><a href="figure-1-cohort-characterization.html#cb36-3" aria-hidden="true" tabindex="-1"></a>pheno<span class="ot">&lt;-</span><span class="fu">readRDS</span>(<span class="at">file=</span><span class="st">&quot;./data/pheno.rds&quot;</span>)</span></code></pre></div>
<p>One filter we put in was to exclude samples &lt;100 cells.</p>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="figure-1-cohort-characterization.html#cb37-1" aria-hidden="true" tabindex="-1"></a>high_quality_samples<span class="ot">&lt;-</span><span class="fu">names</span>(final_NGTs)[<span class="fu">sapply</span>(<span class="fu">names</span>(final_NGTs),<span class="cf">function</span>(x){</span>
<span id="cb37-2"><a href="figure-1-cohort-characterization.html#cb37-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">nrow</span>(final_NGTs[[x]])<span class="sc">&gt;</span><span class="dv">100</span></span>
<span id="cb37-3"><a href="figure-1-cohort-characterization.html#cb37-3" aria-hidden="true" tabindex="-1"></a>})]</span></code></pre></div>
<p>Next we want a catalogue of all mutations so we can determine how many patients were mutated for each gene and how many different mutations were seen in total for each gene.</p>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="figure-1-cohort-characterization.html#cb38-1" aria-hidden="true" tabindex="-1"></a>final_mut_melt<span class="ot">&lt;-</span><span class="fu">do.call</span>(rbind,<span class="fu">lapply</span>(<span class="fu">names</span>(final_NGTs),<span class="cf">function</span>(x){</span>
<span id="cb38-2"><a href="figure-1-cohort-characterization.html#cb38-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.frame</span>(<span class="st">&quot;Sample&quot;</span><span class="ot">=</span>x,</span>
<span id="cb38-3"><a href="figure-1-cohort-characterization.html#cb38-3" aria-hidden="true" tabindex="-1"></a>             <span class="st">&quot;Mutation&quot;</span><span class="ot">=</span><span class="fu">colnames</span>(final_NGTs[[x]]),</span>
<span id="cb38-4"><a href="figure-1-cohort-characterization.html#cb38-4" aria-hidden="true" tabindex="-1"></a>             <span class="st">&quot;Gene&quot;</span><span class="ot">=</span><span class="fu">do.call</span>(rbind,<span class="fu">strsplit</span>(<span class="fu">colnames</span>(final_NGTs[[x]]),<span class="at">split=</span><span class="st">&quot;[:_]&quot;</span>))[,<span class="dv">1</span>])</span>
<span id="cb38-5"><a href="figure-1-cohort-characterization.html#cb38-5" aria-hidden="true" tabindex="-1"></a>      }))</span></code></pre></div>
<p>Next pages will go into how we made each of the Figures.</p>
<div id="cohort-characteristics-ef1a-d" class="section level3" number="3.1.1">
<h3><span class="header-section-number">3.1.1</span> Cohort characteristics (EF1A-D)</h3>
<p>Now we can start to make the plots that are in Extended Figure 1. Starting with the total number of mutations identified per gene</p>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="figure-1-cohort-characterization.html#cb39-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Set the levels of the Gene column from most to least prevalent for plotting purposes</span></span>
<span id="cb39-2"><a href="figure-1-cohort-characterization.html#cb39-2" aria-hidden="true" tabindex="-1"></a>final_mut_melt<span class="sc">$</span>Gene<span class="ot">&lt;-</span> <span class="fu">factor</span>(final_mut_melt<span class="sc">$</span>Gene,<span class="at">levels=</span><span class="fu">names</span>(<span class="fu">sort</span>(<span class="fu">table</span>(final_mut_melt<span class="sc">$</span>Gene), <span class="at">decreasing=</span><span class="cn">TRUE</span>)))</span>
<span id="cb39-3"><a href="figure-1-cohort-characterization.html#cb39-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-4"><a href="figure-1-cohort-characterization.html#cb39-4" aria-hidden="true" tabindex="-1"></a>gg_mut_count<span class="ot">&lt;-</span><span class="fu">ggplot</span>(final_mut_melt,<span class="fu">aes</span>(<span class="at">x=</span>Gene))<span class="sc">+</span></span>
<span id="cb39-5"><a href="figure-1-cohort-characterization.html#cb39-5" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">&quot;count&quot;</span>)<span class="sc">+</span></span>
<span id="cb39-6"><a href="figure-1-cohort-characterization.html#cb39-6" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">theme_classic</span>(<span class="at">base_size =</span> <span class="dv">10</span>)<span class="sc">+</span></span>
<span id="cb39-7"><a href="figure-1-cohort-characterization.html#cb39-7" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">ylab</span>(<span class="st">&quot;Count&quot;</span>)<span class="sc">+</span></span>
<span id="cb39-8"><a href="figure-1-cohort-characterization.html#cb39-8" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">ggtitle</span>(<span class="st">&quot;Number of mutations&quot;</span>)<span class="sc">+</span></span>
<span id="cb39-9"><a href="figure-1-cohort-characterization.html#cb39-9" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle=</span><span class="dv">45</span>, <span class="at">hjust=</span><span class="dv">1</span>,<span class="at">vjust=</span><span class="dv">1</span>),</span>
<span id="cb39-10"><a href="figure-1-cohort-characterization.html#cb39-10" aria-hidden="true" tabindex="-1"></a>                              <span class="at">plot.title=</span><span class="fu">element_text</span>(<span class="at">hjust=</span><span class="fl">0.5</span>))<span class="sc">+</span></span>
<span id="cb39-11"><a href="figure-1-cohort-characterization.html#cb39-11" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">scale_y_continuous</span>(<span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>))</span></code></pre></div>
<p>Total number of patients mutated for each gene</p>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="figure-1-cohort-characterization.html#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="do">## tally of how many mutations per patient</span></span>
<span id="cb40-2"><a href="figure-1-cohort-characterization.html#cb40-2" aria-hidden="true" tabindex="-1"></a>melted_mut_mat<span class="ot">&lt;-</span> final_mut_melt<span class="sc">%&gt;%</span>dplyr<span class="sc">::</span><span class="fu">count</span>(Gene, Sample)</span>
<span id="cb40-3"><a href="figure-1-cohort-characterization.html#cb40-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-4"><a href="figure-1-cohort-characterization.html#cb40-4" aria-hidden="true" tabindex="-1"></a><span class="do">## Set the levels of the Gene column from most to least prevalent for plotting purposes</span></span>
<span id="cb40-5"><a href="figure-1-cohort-characterization.html#cb40-5" aria-hidden="true" tabindex="-1"></a>melted_mut_mat<span class="sc">$</span>Gene<span class="ot">&lt;-</span> <span class="fu">factor</span>(melted_mut_mat<span class="sc">$</span>Gene,<span class="at">levels=</span><span class="fu">names</span>(<span class="fu">sort</span>(<span class="fu">table</span>(melted_mut_mat<span class="sc">$</span>Gene),<span class="at">decreasing=</span><span class="cn">TRUE</span>)))</span>
<span id="cb40-6"><a href="figure-1-cohort-characterization.html#cb40-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-7"><a href="figure-1-cohort-characterization.html#cb40-7" aria-hidden="true" tabindex="-1"></a>gg_mut_patient<span class="ot">&lt;-</span><span class="fu">ggplot</span>(melted_mut_mat,<span class="fu">aes</span>(<span class="at">x=</span>Gene))<span class="sc">+</span></span>
<span id="cb40-8"><a href="figure-1-cohort-characterization.html#cb40-8" aria-hidden="true" tabindex="-1"></a>                      <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">&quot;count&quot;</span>)<span class="sc">+</span></span>
<span id="cb40-9"><a href="figure-1-cohort-characterization.html#cb40-9" aria-hidden="true" tabindex="-1"></a>                      <span class="fu">theme_classic</span>(<span class="at">base_size =</span><span class="dv">10</span>)<span class="sc">+</span></span>
<span id="cb40-10"><a href="figure-1-cohort-characterization.html#cb40-10" aria-hidden="true" tabindex="-1"></a>                      <span class="fu">ylab</span>(<span class="st">&quot;Count&quot;</span>)<span class="sc">+</span><span class="fu">ggtitle</span>(<span class="st">&quot;Number of patients with mutation&quot;</span>)<span class="sc">+</span></span>
<span id="cb40-11"><a href="figure-1-cohort-characterization.html#cb40-11" aria-hidden="true" tabindex="-1"></a>                      <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle=</span><span class="dv">45</span>,<span class="at">hjust=</span><span class="dv">1</span>,<span class="at">vjust=</span><span class="dv">1</span>),</span>
<span id="cb40-12"><a href="figure-1-cohort-characterization.html#cb40-12" aria-hidden="true" tabindex="-1"></a>                            <span class="at">plot.title=</span><span class="fu">element_text</span>(<span class="at">hjust=</span><span class="fl">0.5</span>))<span class="sc">+</span></span>
<span id="cb40-13"><a href="figure-1-cohort-characterization.html#cb40-13" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">scale_y_continuous</span>(<span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>))</span></code></pre></div>
<p>Number of mutated genes per patient</p>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="figure-1-cohort-characterization.html#cb41-1" aria-hidden="true" tabindex="-1"></a>gg_mutated_genes_per_patient<span class="ot">&lt;-</span>final_mut_melt<span class="sc">%&gt;%</span></span>
<span id="cb41-2"><a href="figure-1-cohort-characterization.html#cb41-2" aria-hidden="true" tabindex="-1"></a>                                          <span class="fu">distinct</span>(Sample,Gene)<span class="sc">%&gt;%</span></span>
<span id="cb41-3"><a href="figure-1-cohort-characterization.html#cb41-3" aria-hidden="true" tabindex="-1"></a>                                          <span class="fu">group_by</span>(Sample)<span class="sc">%&gt;%</span></span>
<span id="cb41-4"><a href="figure-1-cohort-characterization.html#cb41-4" aria-hidden="true" tabindex="-1"></a>                                          tally<span class="sc">%&gt;%</span></span>
<span id="cb41-5"><a href="figure-1-cohort-characterization.html#cb41-5" aria-hidden="true" tabindex="-1"></a>                                          <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>n))<span class="sc">+</span><span class="fu">geom_bar</span>()<span class="sc">+</span></span>
<span id="cb41-6"><a href="figure-1-cohort-characterization.html#cb41-6" aria-hidden="true" tabindex="-1"></a>                                          <span class="fu">ylab</span>(<span class="st">&quot;Count&quot;</span>)<span class="sc">+</span></span>
<span id="cb41-7"><a href="figure-1-cohort-characterization.html#cb41-7" aria-hidden="true" tabindex="-1"></a>                                          <span class="fu">xlab</span>(<span class="st">&quot;Number of genes&quot;</span>)<span class="sc">+</span></span>
<span id="cb41-8"><a href="figure-1-cohort-characterization.html#cb41-8" aria-hidden="true" tabindex="-1"></a>                                          <span class="fu">ggtitle</span>(<span class="st">&quot;Mutant genes per patient&quot;</span>)<span class="sc">+</span></span>
<span id="cb41-9"><a href="figure-1-cohort-characterization.html#cb41-9" aria-hidden="true" tabindex="-1"></a>                                          <span class="fu">theme_classic</span>(<span class="at">base_size =</span> <span class="dv">10</span>)<span class="sc">+</span></span>
<span id="cb41-10"><a href="figure-1-cohort-characterization.html#cb41-10" aria-hidden="true" tabindex="-1"></a>                                          <span class="fu">theme</span>(<span class="at">plot.title=</span><span class="fu">element_text</span>(<span class="at">hjust=</span><span class="fl">0.5</span>))<span class="sc">+</span></span>
<span id="cb41-11"><a href="figure-1-cohort-characterization.html#cb41-11" aria-hidden="true" tabindex="-1"></a>                                          <span class="fu">scale_y_continuous</span>(<span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>))<span class="sc">+</span></span>
<span id="cb41-12"><a href="figure-1-cohort-characterization.html#cb41-12" aria-hidden="true" tabindex="-1"></a>                                          <span class="fu">scale_x_continuous</span>(<span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>),<span class="at">n.breaks=</span><span class="dv">8</span>)</span></code></pre></div>
<p>Total number of mutations per patient</p>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="figure-1-cohort-characterization.html#cb42-1" aria-hidden="true" tabindex="-1"></a>gg_mutations_per_patient<span class="ot">&lt;-</span> final_mut_melt<span class="sc">%&gt;%</span></span>
<span id="cb42-2"><a href="figure-1-cohort-characterization.html#cb42-2" aria-hidden="true" tabindex="-1"></a>                                        <span class="fu">group_by</span>(Sample)<span class="sc">%&gt;%</span></span>
<span id="cb42-3"><a href="figure-1-cohort-characterization.html#cb42-3" aria-hidden="true" tabindex="-1"></a>                                        tally<span class="sc">%&gt;%</span></span>
<span id="cb42-4"><a href="figure-1-cohort-characterization.html#cb42-4" aria-hidden="true" tabindex="-1"></a>                                        <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>n))<span class="sc">+</span><span class="fu">geom_bar</span>()<span class="sc">+</span></span>
<span id="cb42-5"><a href="figure-1-cohort-characterization.html#cb42-5" aria-hidden="true" tabindex="-1"></a>                                        <span class="fu">theme_classic</span>(<span class="at">base_size =</span> <span class="dv">10</span>)<span class="sc">+</span></span>
<span id="cb42-6"><a href="figure-1-cohort-characterization.html#cb42-6" aria-hidden="true" tabindex="-1"></a>                                        <span class="fu">theme</span>(<span class="at">plot.title=</span><span class="fu">element_text</span>(<span class="at">hjust=</span><span class="fl">0.5</span>))<span class="sc">+</span></span>
<span id="cb42-7"><a href="figure-1-cohort-characterization.html#cb42-7" aria-hidden="true" tabindex="-1"></a>                                        <span class="fu">ylab</span>(<span class="st">&quot;Count&quot;</span>)<span class="sc">+</span><span class="fu">ggtitle</span>(<span class="st">&quot;Variants per patient&quot;</span>)<span class="sc">+</span><span class="fu">xlab</span>(<span class="st">&quot;Number of variants&quot;</span>)<span class="sc">+</span>                              </span>
<span id="cb42-8"><a href="figure-1-cohort-characterization.html#cb42-8" aria-hidden="true" tabindex="-1"></a>                                        <span class="fu">scale_y_continuous</span>(<span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>))<span class="sc">+</span></span>
<span id="cb42-9"><a href="figure-1-cohort-characterization.html#cb42-9" aria-hidden="true" tabindex="-1"></a>                                        <span class="fu">scale_x_continuous</span>(<span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>),<span class="at">n.breaks=</span><span class="dv">6</span>)</span></code></pre></div>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="figure-1-cohort-characterization.html#cb43-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(cowplot)</span>
<span id="cb43-2"><a href="figure-1-cohort-characterization.html#cb43-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_grid</span>(gg_mut_count,gg_mut_patient,</span>
<span id="cb43-3"><a href="figure-1-cohort-characterization.html#cb43-3" aria-hidden="true" tabindex="-1"></a>          gg_mutated_genes_per_patient,gg_mutations_per_patient,</span>
<span id="cb43-4"><a href="figure-1-cohort-characterization.html#cb43-4" aria-hidden="true" tabindex="-1"></a>          <span class="at">ncol=</span><span class="dv">2</span>,<span class="at">align=</span><span class="st">&quot;hv&quot;</span>,<span class="at">axis=</span><span class="st">&quot;ltrb&quot;</span>,</span>
<span id="cb43-5"><a href="figure-1-cohort-characterization.html#cb43-5" aria-hidden="true" tabindex="-1"></a>          <span class="at">labels =</span> <span class="st">&quot;AUTO&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span id="fig:unnamed-chunk-21"></span>
<img src="scDNA_files/figure-html/unnamed-chunk-21-1.png" alt="Extended Figure 1A-D" width="864" />
<p class="caption">
Figure 3.1: Extended Figure 1A-D
</p>
</div>
<p>An extra plot worth noting is below, which plots the number of cells per sample, after we filterd out all of the “3” uniformative genotyped cells.</p>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="figure-1-cohort-characterization.html#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="do">### Cells per sample</span></span>
<span id="cb44-2"><a href="figure-1-cohort-characterization.html#cb44-2" aria-hidden="true" tabindex="-1"></a><span class="fu">data.frame</span>(<span class="st">&quot;Cells&quot;</span><span class="ot">=</span><span class="fu">do.call</span>(rbind,<span class="fu">lapply</span>(final_NGTs,nrow)))<span class="sc">%&gt;%</span></span>
<span id="cb44-3"><a href="figure-1-cohort-characterization.html#cb44-3" aria-hidden="true" tabindex="-1"></a>                              <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>Cells))<span class="sc">+</span><span class="fu">geom_histogram</span>(<span class="at">binwidth =</span> <span class="dv">100</span>)<span class="sc">+</span></span>
<span id="cb44-4"><a href="figure-1-cohort-characterization.html#cb44-4" aria-hidden="true" tabindex="-1"></a>                                      <span class="fu">theme_classic</span>(<span class="at">base_size =</span> <span class="dv">10</span>)<span class="sc">+</span></span>
<span id="cb44-5"><a href="figure-1-cohort-characterization.html#cb44-5" aria-hidden="true" tabindex="-1"></a>                                      <span class="fu">theme</span>(<span class="at">plot.title=</span><span class="fu">element_text</span>(<span class="at">hjust=</span><span class="fl">0.5</span>))<span class="sc">+</span></span>
<span id="cb44-6"><a href="figure-1-cohort-characterization.html#cb44-6" aria-hidden="true" tabindex="-1"></a>                                      <span class="fu">ylab</span>(<span class="st">&quot;Count&quot;</span>)<span class="sc">+</span><span class="fu">ggtitle</span>(<span class="st">&quot;Informative Cells per Sample&quot;</span>)<span class="sc">+</span></span>
<span id="cb44-7"><a href="figure-1-cohort-characterization.html#cb44-7" aria-hidden="true" tabindex="-1"></a>                                      <span class="fu">scale_y_continuous</span>(<span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>))<span class="sc">+</span></span>
<span id="cb44-8"><a href="figure-1-cohort-characterization.html#cb44-8" aria-hidden="true" tabindex="-1"></a>                                      <span class="fu">scale_x_continuous</span>(<span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>),<span class="at">n.breaks=</span><span class="dv">8</span>)</span></code></pre></div>
<p><img src="scDNA_files/figure-html/unnamed-chunk-22-1.png" width="384" style="display: block; margin: auto;" /></p>
</div>
<div id="mutation-co-occurence" class="section level3" number="3.1.2">
<h3><span class="header-section-number">3.1.2</span> Mutation Co-occurence</h3>
<p>Next we want to make the co-occurence matrix on a sample level</p>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="figure-1-cohort-characterization.html#cb45-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(cooccur)</span>
<span id="cb45-2"><a href="figure-1-cohort-characterization.html#cb45-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-3"><a href="figure-1-cohort-characterization.html#cb45-3" aria-hidden="true" tabindex="-1"></a><span class="do">### create matrix for oncoprint</span></span>
<span id="cb45-4"><a href="figure-1-cohort-characterization.html#cb45-4" aria-hidden="true" tabindex="-1"></a>mut_mat <span class="ot">&lt;-</span> <span class="fu">table</span>(melted_mut_mat<span class="sc">$</span>Sample,melted_mut_mat<span class="sc">$</span>Gene)</span>
<span id="cb45-5"><a href="figure-1-cohort-characterization.html#cb45-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-6"><a href="figure-1-cohort-characterization.html#cb45-6" aria-hidden="true" tabindex="-1"></a><span class="do">### Prepare matrix for co occurence map</span></span>
<span id="cb45-7"><a href="figure-1-cohort-characterization.html#cb45-7" aria-hidden="true" tabindex="-1"></a>cooccur_mat <span class="ot">&lt;-</span> <span class="fu">cooccur</span>(<span class="at">mat=</span><span class="fu">t</span>(mut_mat), <span class="at">type=</span><span class="st">&quot;spp_site&quot;</span>,</span>
<span id="cb45-8"><a href="figure-1-cohort-characterization.html#cb45-8" aria-hidden="true" tabindex="-1"></a>                       <span class="at">only_effects =</span> <span class="cn">FALSE</span>,<span class="at">eff_matrix=</span><span class="cn">TRUE</span>,</span>
<span id="cb45-9"><a href="figure-1-cohort-characterization.html#cb45-9" aria-hidden="true" tabindex="-1"></a>                       <span class="at">thresh=</span><span class="cn">FALSE</span>, <span class="at">eff_standard=</span><span class="cn">FALSE</span>,<span class="at">spp_names=</span><span class="cn">TRUE</span>)<span class="sc">$</span>results</span>
<span id="cb45-10"><a href="figure-1-cohort-characterization.html#cb45-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-11"><a href="figure-1-cohort-characterization.html#cb45-11" aria-hidden="true" tabindex="-1"></a><span class="do">## Denote which interactions are significantly inclusive or exclusive </span></span>
<span id="cb45-12"><a href="figure-1-cohort-characterization.html#cb45-12" aria-hidden="true" tabindex="-1"></a><span class="co"># The &#39;add_row&#39; function generates a new line, but it gets removed later.</span></span>
<span id="cb45-13"><a href="figure-1-cohort-characterization.html#cb45-13" aria-hidden="true" tabindex="-1"></a><span class="co"># This is helpful for setting the order of the gene labels below. </span></span>
<span id="cb45-14"><a href="figure-1-cohort-characterization.html#cb45-14" aria-hidden="true" tabindex="-1"></a>cooccur_data_mat <span class="ot">&lt;-</span> cooccur_mat<span class="sc">%&gt;%</span></span>
<span id="cb45-15"><a href="figure-1-cohort-characterization.html#cb45-15" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">mutate</span>(<span class="at">score=</span><span class="fu">ifelse</span>(p_lt<span class="sc">&lt;=</span><span class="fl">0.05</span>,<span class="sc">-</span><span class="dv">1</span>,</span>
<span id="cb45-16"><a href="figure-1-cohort-characterization.html#cb45-16" aria-hidden="true" tabindex="-1"></a>                                            <span class="fu">ifelse</span>(p_gt<span class="sc">&lt;=</span><span class="fl">0.05</span>,<span class="dv">1</span>,<span class="dv">0</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb45-17"><a href="figure-1-cohort-characterization.html#cb45-17" aria-hidden="true" tabindex="-1"></a>                        dplyr<span class="sc">::</span><span class="fu">select</span>(sp1_name,sp2_name,score)<span class="sc">%&gt;%</span></span>
<span id="cb45-18"><a href="figure-1-cohort-characterization.html#cb45-18" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">add_row</span>(<span class="at">sp2_name=</span><span class="fu">setdiff</span>(.<span class="sc">$</span>sp1_name,.<span class="sc">$</span>sp2_name),</span>
<span id="cb45-19"><a href="figure-1-cohort-characterization.html#cb45-19" aria-hidden="true" tabindex="-1"></a>                                <span class="at">sp1_name=</span><span class="fu">setdiff</span>(.<span class="sc">$</span>sp2_name,.<span class="sc">$</span>sp1_name),</span>
<span id="cb45-20"><a href="figure-1-cohort-characterization.html#cb45-20" aria-hidden="true" tabindex="-1"></a>                                <span class="at">score=</span><span class="dv">0</span>)</span>
<span id="cb45-21"><a href="figure-1-cohort-characterization.html#cb45-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-22"><a href="figure-1-cohort-characterization.html#cb45-22" aria-hidden="true" tabindex="-1"></a><span class="co">#check out the final that we added so we can remove it later</span></span>
<span id="cb45-23"><a href="figure-1-cohort-characterization.html#cb45-23" aria-hidden="true" tabindex="-1"></a><span class="fu">tail</span>(cooccur_data_mat) </span>
<span id="cb45-24"><a href="figure-1-cohort-characterization.html#cb45-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-25"><a href="figure-1-cohort-characterization.html#cb45-25" aria-hidden="true" tabindex="-1"></a><span class="co"># Order the genes in a coherent pattern for triangle strucutre of graph.</span></span>
<span id="cb45-26"><a href="figure-1-cohort-characterization.html#cb45-26" aria-hidden="true" tabindex="-1"></a>cooccur_data_mat<span class="sc">$</span>sp1_name<span class="ot">&lt;-</span><span class="fu">factor</span>(cooccur_data_mat<span class="sc">$</span>sp1_name,</span>
<span id="cb45-27"><a href="figure-1-cohort-characterization.html#cb45-27" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">levels=</span><span class="fu">unique</span>(cooccur_data_mat<span class="sc">$</span>sp1_name))</span>
<span id="cb45-28"><a href="figure-1-cohort-characterization.html#cb45-28" aria-hidden="true" tabindex="-1"></a>cooccur_data_mat<span class="sc">$</span>sp2_name<span class="ot">&lt;-</span><span class="fu">factor</span>(cooccur_data_mat<span class="sc">$</span>sp2_name,</span>
<span id="cb45-29"><a href="figure-1-cohort-characterization.html#cb45-29" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">levels=</span><span class="fu">rev</span>(<span class="fu">levels</span>(cooccur_data_mat<span class="sc">$</span>sp1_name)))</span>
<span id="cb45-30"><a href="figure-1-cohort-characterization.html#cb45-30" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb45-31"><a href="figure-1-cohort-characterization.html#cb45-31" aria-hidden="true" tabindex="-1"></a><span class="co"># Triangle heatmap to compare cohorts</span></span>
<span id="cb45-32"><a href="figure-1-cohort-characterization.html#cb45-32" aria-hidden="true" tabindex="-1"></a>grob_corrplot<span class="ot">&lt;-</span><span class="fu">ggplot</span>(cooccur_data_mat<span class="sc">%&gt;%</span><span class="fu">filter</span>(sp1_name<span class="sc">!=</span><span class="st">&quot;BRAF&quot;</span>),<span class="fu">aes</span>(<span class="at">x=</span>sp1_name,<span class="at">y=</span>sp2_name))<span class="sc">+</span></span>
<span id="cb45-33"><a href="figure-1-cohort-characterization.html#cb45-33" aria-hidden="true" tabindex="-1"></a>                      <span class="fu">geom_tile</span>(<span class="fu">aes</span>(<span class="at">fill =</span> <span class="fu">factor</span>(score)), <span class="at">color=</span><span class="st">&#39;grey90&#39;</span>) <span class="sc">+</span></span>
<span id="cb45-34"><a href="figure-1-cohort-characterization.html#cb45-34" aria-hidden="true" tabindex="-1"></a>                      <span class="fu">scale_fill_manual</span>(<span class="at">name=</span><span class="st">&quot;Correlation&quot;</span>,</span>
<span id="cb45-35"><a href="figure-1-cohort-characterization.html#cb45-35" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">values=</span><span class="fu">c</span>(<span class="st">&quot;-1&quot;</span><span class="ot">=</span><span class="st">&quot;firebrick3&quot;</span>,</span>
<span id="cb45-36"><a href="figure-1-cohort-characterization.html#cb45-36" aria-hidden="true" tabindex="-1"></a>                                                 <span class="st">&quot;0&quot;</span><span class="ot">=</span><span class="st">&quot;white&quot;</span>,</span>
<span id="cb45-37"><a href="figure-1-cohort-characterization.html#cb45-37" aria-hidden="true" tabindex="-1"></a>                                                 <span class="st">&quot;1&quot;</span><span class="ot">=</span><span class="st">&quot;steelblue2&quot;</span>),</span>
<span id="cb45-38"><a href="figure-1-cohort-characterization.html#cb45-38" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">labels=</span><span class="fu">c</span>(<span class="st">&quot;Mutually Exclusive&quot;</span>,</span>
<span id="cb45-39"><a href="figure-1-cohort-characterization.html#cb45-39" aria-hidden="true" tabindex="-1"></a>                                                 <span class="st">&quot;Not Significant&quot;</span>,</span>
<span id="cb45-40"><a href="figure-1-cohort-characterization.html#cb45-40" aria-hidden="true" tabindex="-1"></a>                                                 <span class="st">&quot;Mutually Inclusive&quot;</span>))<span class="sc">+</span></span>
<span id="cb45-41"><a href="figure-1-cohort-characterization.html#cb45-41" aria-hidden="true" tabindex="-1"></a>                      <span class="fu">theme_classic</span>(<span class="at">base_size=</span><span class="dv">10</span>)<span class="sc">+</span></span>
<span id="cb45-42"><a href="figure-1-cohort-characterization.html#cb45-42" aria-hidden="true" tabindex="-1"></a>                      <span class="fu">xlab</span>(<span class="st">&quot;&quot;</span>)<span class="sc">+</span><span class="fu">ylab</span>(<span class="st">&quot;&quot;</span>)<span class="sc">+</span></span>
<span id="cb45-43"><a href="figure-1-cohort-characterization.html#cb45-43" aria-hidden="true" tabindex="-1"></a>                      <span class="fu">theme</span>(<span class="at">axis.text.x=</span><span class="fu">element_text</span>(<span class="at">angle=</span><span class="dv">45</span>,<span class="at">hjust=</span><span class="dv">1</span>,<span class="at">vjust=</span><span class="dv">1</span>),</span>
<span id="cb45-44"><a href="figure-1-cohort-characterization.html#cb45-44" aria-hidden="true" tabindex="-1"></a>                            <span class="at">axis.line =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb45-45"><a href="figure-1-cohort-characterization.html#cb45-45" aria-hidden="true" tabindex="-1"></a>                            <span class="at">legend.position =</span> <span class="fu">c</span>(<span class="fl">0.8</span>,<span class="dv">1</span>), </span>
<span id="cb45-46"><a href="figure-1-cohort-characterization.html#cb45-46" aria-hidden="true" tabindex="-1"></a>                            <span class="at">legend.justification =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">1</span>),</span>
<span id="cb45-47"><a href="figure-1-cohort-characterization.html#cb45-47" aria-hidden="true" tabindex="-1"></a>                            <span class="at">legend.direction =</span> <span class="st">&quot;vertical&quot;</span>)<span class="sc">+</span></span>
<span id="cb45-48"><a href="figure-1-cohort-characterization.html#cb45-48" aria-hidden="true" tabindex="-1"></a>                      <span class="fu">theme</span>(<span class="at">legend.key.size =</span> <span class="fu">unit</span>(<span class="fl">0.5</span>,<span class="st">&quot;line&quot;</span>))</span>
<span id="cb45-49"><a href="figure-1-cohort-characterization.html#cb45-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-50"><a href="figure-1-cohort-characterization.html#cb45-50" aria-hidden="true" tabindex="-1"></a>grob_corrplot</span></code></pre></div>
<p><img src="scDNA_files/figure-html/unnamed-chunk-23-1.png" width="480" style="display: block; margin: auto;" /></p>

</div>
<div id="oncoprint-figure-1a" class="section level3" number="3.1.3">
<h3><span class="header-section-number">3.1.3</span> Oncoprint: Figure 1A</h3>
<p>Now we are going to make the oncoprint that is in Figure 1A. Check out this <a href="https://jokergoo.github.io/ComplexHeatmap-reference/book/oncoprint.html">example</a> from the ComplexHeatmap package for a better understanding of why things are formatted the way they are.</p>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="figure-1-cohort-characterization.html#cb46-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(pals) <span class="co"># great package with color palettes in R</span></span>
<span id="cb46-2"><a href="figure-1-cohort-characterization.html#cb46-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ComplexHeatmap) <span class="co">#used for making the oncoprint</span></span>
<span id="cb46-3"><a href="figure-1-cohort-characterization.html#cb46-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-4"><a href="figure-1-cohort-characterization.html#cb46-4" aria-hidden="true" tabindex="-1"></a><span class="co">#Here we will identify whether a variant is an indel, Nonsense or Missense mutation</span></span>
<span id="cb46-5"><a href="figure-1-cohort-characterization.html#cb46-5" aria-hidden="true" tabindex="-1"></a><span class="co">#Next we group samples together and pivot the matrix into wide format</span></span>
<span id="cb46-6"><a href="figure-1-cohort-characterization.html#cb46-6" aria-hidden="true" tabindex="-1"></a>fill_values <span class="ot">&lt;-</span><span class="fu">setNames</span>(<span class="fu">as.list</span>(<span class="fu">rep</span>(<span class="dv">0</span>,<span class="fu">length</span>(<span class="fu">levels</span>(final_mut_melt<span class="sc">$</span>Gene)))),<span class="fu">levels</span>(final_mut_melt<span class="sc">$</span>Gene))</span>
<span id="cb46-7"><a href="figure-1-cohort-characterization.html#cb46-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-8"><a href="figure-1-cohort-characterization.html#cb46-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-9"><a href="figure-1-cohort-characterization.html#cb46-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-10"><a href="figure-1-cohort-characterization.html#cb46-10" aria-hidden="true" tabindex="-1"></a>mut_mat_wide<span class="ot">&lt;-</span>final_mut_melt<span class="sc">%&gt;%</span></span>
<span id="cb46-11"><a href="figure-1-cohort-characterization.html#cb46-11" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">mutate</span>(<span class="at">Variant_Class=</span><span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">&quot;fs</span><span class="sc">\\</span><span class="st">*|INS_|ins|ext|del&quot;</span>,.<span class="sc">$</span>Mutation),<span class="st">&quot;Indel&quot;</span>,</span>
<span id="cb46-12"><a href="figure-1-cohort-characterization.html#cb46-12" aria-hidden="true" tabindex="-1"></a>                                                  <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">&quot;</span><span class="sc">\\</span><span class="st">*$&quot;</span>,.<span class="sc">$</span>Mutation),<span class="st">&quot;Nonsense&quot;</span>,<span class="st">&quot;Missense&quot;</span>)))<span class="sc">%&gt;%</span></span>
<span id="cb46-13"><a href="figure-1-cohort-characterization.html#cb46-13" aria-hidden="true" tabindex="-1"></a>                      <span class="co">#  mutate_at(c(&quot;Gene&quot;,&quot;Variant_Class&quot;,&quot;Sample&quot;),as.character())%&gt;%</span></span>
<span id="cb46-14"><a href="figure-1-cohort-characterization.html#cb46-14" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">group_by</span>(Sample,Gene)<span class="sc">%&gt;%</span></span>
<span id="cb46-15"><a href="figure-1-cohort-characterization.html#cb46-15" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">summarize</span>(<span class="st">&quot;Variants&quot;</span><span class="ot">=</span><span class="fu">list</span>(Variant_Class))<span class="sc">%&gt;%</span></span>
<span id="cb46-16"><a href="figure-1-cohort-characterization.html#cb46-16" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">pivot_wider</span>(<span class="at">id_cols=</span>Sample, </span>
<span id="cb46-17"><a href="figure-1-cohort-characterization.html#cb46-17" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">names_from  =</span> Gene, </span>
<span id="cb46-18"><a href="figure-1-cohort-characterization.html#cb46-18" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">values_from =</span> Variants,</span>
<span id="cb46-19"><a href="figure-1-cohort-characterization.html#cb46-19" aria-hidden="true" tabindex="-1"></a>                                   <span class="co">#  values_fn   = list(Variant_Class=list),</span></span>
<span id="cb46-20"><a href="figure-1-cohort-characterization.html#cb46-20" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">values_fill =</span> <span class="fu">list</span>(<span class="at">Variant_Class=</span><span class="st">&quot;&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb46-21"><a href="figure-1-cohort-characterization.html#cb46-21" aria-hidden="true" tabindex="-1"></a>                        <span class="co">#ungroup(Sample,Gene)%&gt;%</span></span>
<span id="cb46-22"><a href="figure-1-cohort-characterization.html#cb46-22" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">data.frame</span>()</span>
<span id="cb46-23"><a href="figure-1-cohort-characterization.html#cb46-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-24"><a href="figure-1-cohort-characterization.html#cb46-24" aria-hidden="true" tabindex="-1"></a><span class="co"># At this point, each column is actually a list, and variants are represnted as a list within a list. </span></span>
<span id="cb46-25"><a href="figure-1-cohort-characterization.html#cb46-25" aria-hidden="true" tabindex="-1"></a><span class="co"># So we want to unpack it a bit, and turn those variant lists in a ; separated vector</span></span>
<span id="cb46-26"><a href="figure-1-cohort-characterization.html#cb46-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-27"><a href="figure-1-cohort-characterization.html#cb46-27" aria-hidden="true" tabindex="-1"></a>mut_mat_wide_storage <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb46-28"><a href="figure-1-cohort-characterization.html#cb46-28" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">2</span><span class="sc">:</span><span class="fu">ncol</span>(mut_mat_wide)){ <span class="co"># start at 2 to ignore the first column of sample names</span></span>
<span id="cb46-29"><a href="figure-1-cohort-characterization.html#cb46-29" aria-hidden="true" tabindex="-1"></a>  mut_mat_wide_storage[[i]]<span class="ot">&lt;-</span> <span class="fu">do.call</span>(rbind,<span class="fu">lapply</span>(mut_mat_wide[,i],<span class="cf">function</span>(x){</span>
<span id="cb46-30"><a href="figure-1-cohort-characterization.html#cb46-30" aria-hidden="true" tabindex="-1"></a>    <span class="co">#if(x==&quot;&quot;){</span></span>
<span id="cb46-31"><a href="figure-1-cohort-characterization.html#cb46-31" aria-hidden="true" tabindex="-1"></a>    <span class="co">#  return(x)</span></span>
<span id="cb46-32"><a href="figure-1-cohort-characterization.html#cb46-32" aria-hidden="true" tabindex="-1"></a>  <span class="co">#  }  else  </span></span>
<span id="cb46-33"><a href="figure-1-cohort-characterization.html#cb46-33" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span>(<span class="fu">is.null</span>(x)){</span>
<span id="cb46-34"><a href="figure-1-cohort-characterization.html#cb46-34" aria-hidden="true" tabindex="-1"></a>      <span class="fu">return</span>(<span class="st">&quot;&quot;</span>)</span>
<span id="cb46-35"><a href="figure-1-cohort-characterization.html#cb46-35" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb46-36"><a href="figure-1-cohort-characterization.html#cb46-36" aria-hidden="true" tabindex="-1"></a>    <span class="fu">paste</span>(x,<span class="at">sep=</span><span class="st">&quot;;&quot;</span>,<span class="at">collapse=</span><span class="st">&quot;;&quot;</span>)</span>
<span id="cb46-37"><a href="figure-1-cohort-characterization.html#cb46-37" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb46-38"><a href="figure-1-cohort-characterization.html#cb46-38" aria-hidden="true" tabindex="-1"></a>   }</span>
<span id="cb46-39"><a href="figure-1-cohort-characterization.html#cb46-39" aria-hidden="true" tabindex="-1"></a>   ))</span>
<span id="cb46-40"><a href="figure-1-cohort-characterization.html#cb46-40" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb46-41"><a href="figure-1-cohort-characterization.html#cb46-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-42"><a href="figure-1-cohort-characterization.html#cb46-42" aria-hidden="true" tabindex="-1"></a><span class="do">## The rest of this just turns this back into a matrix in the format suggested by ComplexHeatmap</span></span>
<span id="cb46-43"><a href="figure-1-cohort-characterization.html#cb46-43" aria-hidden="true" tabindex="-1"></a>mut_mat_wide_storage[[<span class="dv">1</span>]] <span class="ot">&lt;-</span> <span class="fu">as.character</span>(mut_mat_wide[,<span class="st">&quot;Sample&quot;</span>])</span>
<span id="cb46-44"><a href="figure-1-cohort-characterization.html#cb46-44" aria-hidden="true" tabindex="-1"></a>final_mat <span class="ot">&lt;-</span> <span class="fu">do.call</span>(cbind,mut_mat_wide_storage)</span>
<span id="cb46-45"><a href="figure-1-cohort-characterization.html#cb46-45" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(final_mat) <span class="ot">&lt;-</span> <span class="fu">colnames</span>(mut_mat_wide)</span>
<span id="cb46-46"><a href="figure-1-cohort-characterization.html#cb46-46" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(final_mat) <span class="ot">&lt;-</span> final_mat[,<span class="dv">1</span>]</span>
<span id="cb46-47"><a href="figure-1-cohort-characterization.html#cb46-47" aria-hidden="true" tabindex="-1"></a>final_mat <span class="ot">&lt;-</span> <span class="fu">t</span>(final_mat[,<span class="sc">-</span><span class="dv">1</span>])</span>
<span id="cb46-48"><a href="figure-1-cohort-characterization.html#cb46-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-49"><a href="figure-1-cohort-characterization.html#cb46-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-50"><a href="figure-1-cohort-characterization.html#cb46-50" aria-hidden="true" tabindex="-1"></a><span class="co">#Now we set up the color schemes for the variants on each row</span></span>
<span id="cb46-51"><a href="figure-1-cohort-characterization.html#cb46-51" aria-hidden="true" tabindex="-1"></a>variant_type_colors <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;Indel&quot;</span> <span class="ot">=</span> <span class="st">&quot;darkorchid2&quot;</span>, <span class="st">&quot;Nonsense&quot;</span> <span class="ot">=</span> <span class="st">&quot;black&quot;</span>, <span class="st">&quot;Missense&quot;</span> <span class="ot">=</span> <span class="st">&quot;darkgreen&quot;</span>)</span>
<span id="cb46-52"><a href="figure-1-cohort-characterization.html#cb46-52" aria-hidden="true" tabindex="-1"></a>alter_functions <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb46-53"><a href="figure-1-cohort-characterization.html#cb46-53" aria-hidden="true" tabindex="-1"></a>  <span class="at">background =</span> <span class="cf">function</span>(x, y, w, h) {</span>
<span id="cb46-54"><a href="figure-1-cohort-characterization.html#cb46-54" aria-hidden="true" tabindex="-1"></a>    <span class="fu">grid.rect</span>(x, y, w, h<span class="sc">-</span><span class="fu">unit</span>(<span class="fl">0.25</span>, <span class="st">&quot;mm&quot;</span>), </span>
<span id="cb46-55"><a href="figure-1-cohort-characterization.html#cb46-55" aria-hidden="true" tabindex="-1"></a>              <span class="at">gp =</span> <span class="fu">gpar</span>(<span class="at">fill =</span> <span class="st">&quot;#CCCCCC&quot;</span>, <span class="at">col =</span> <span class="cn">NA</span>))</span>
<span id="cb46-56"><a href="figure-1-cohort-characterization.html#cb46-56" aria-hidden="true" tabindex="-1"></a>  },</span>
<span id="cb46-57"><a href="figure-1-cohort-characterization.html#cb46-57" aria-hidden="true" tabindex="-1"></a>  <span class="at">Indel =</span> <span class="cf">function</span>(x, y, w, h) {</span>
<span id="cb46-58"><a href="figure-1-cohort-characterization.html#cb46-58" aria-hidden="true" tabindex="-1"></a>    <span class="fu">grid.rect</span>(x, y, w<span class="sc">-</span><span class="fu">unit</span>(<span class="fl">0.5</span>, <span class="st">&quot;mm&quot;</span>), h<span class="sc">-</span><span class="fu">unit</span>(<span class="fl">0.5</span>, <span class="st">&quot;mm&quot;</span>), </span>
<span id="cb46-59"><a href="figure-1-cohort-characterization.html#cb46-59" aria-hidden="true" tabindex="-1"></a>              <span class="at">gp =</span> <span class="fu">gpar</span>(<span class="at">fill =</span> variant_type_colors[<span class="st">&quot;Indel&quot;</span>], <span class="at">col =</span> <span class="cn">NA</span>))</span>
<span id="cb46-60"><a href="figure-1-cohort-characterization.html#cb46-60" aria-hidden="true" tabindex="-1"></a>  },</span>
<span id="cb46-61"><a href="figure-1-cohort-characterization.html#cb46-61" aria-hidden="true" tabindex="-1"></a>  <span class="at">Nonsense =</span> <span class="cf">function</span>(x, y, w, h) {</span>
<span id="cb46-62"><a href="figure-1-cohort-characterization.html#cb46-62" aria-hidden="true" tabindex="-1"></a>    <span class="fu">grid.rect</span>(x, y, w<span class="sc">-</span><span class="fu">unit</span>(<span class="fl">0.5</span>, <span class="st">&quot;mm&quot;</span>), h<span class="sc">-</span><span class="fu">unit</span>(<span class="fl">0.5</span>, <span class="st">&quot;mm&quot;</span>), </span>
<span id="cb46-63"><a href="figure-1-cohort-characterization.html#cb46-63" aria-hidden="true" tabindex="-1"></a>              <span class="at">gp =</span> <span class="fu">gpar</span>(<span class="at">fill =</span> variant_type_colors[<span class="st">&quot;Nonsense&quot;</span>], <span class="at">col =</span> <span class="cn">NA</span>))</span>
<span id="cb46-64"><a href="figure-1-cohort-characterization.html#cb46-64" aria-hidden="true" tabindex="-1"></a>  },</span>
<span id="cb46-65"><a href="figure-1-cohort-characterization.html#cb46-65" aria-hidden="true" tabindex="-1"></a>  <span class="at">Missense =</span> <span class="cf">function</span>(x, y, w, h) {</span>
<span id="cb46-66"><a href="figure-1-cohort-characterization.html#cb46-66" aria-hidden="true" tabindex="-1"></a>    <span class="fu">grid.rect</span>(x, y, w<span class="sc">-</span><span class="fu">unit</span>(<span class="fl">0.5</span>, <span class="st">&quot;mm&quot;</span>), h<span class="sc">*</span><span class="fl">0.33</span>, </span>
<span id="cb46-67"><a href="figure-1-cohort-characterization.html#cb46-67" aria-hidden="true" tabindex="-1"></a>              <span class="at">gp =</span> <span class="fu">gpar</span>(<span class="at">fill =</span> variant_type_colors[<span class="st">&quot;Missense&quot;</span>], <span class="at">col =</span> <span class="cn">NA</span>))</span>
<span id="cb46-68"><a href="figure-1-cohort-characterization.html#cb46-68" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb46-69"><a href="figure-1-cohort-characterization.html#cb46-69" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb46-70"><a href="figure-1-cohort-characterization.html#cb46-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-71"><a href="figure-1-cohort-characterization.html#cb46-71" aria-hidden="true" tabindex="-1"></a><span class="co"># Establish colors for each disease state and diagnosis</span></span>
<span id="cb46-72"><a href="figure-1-cohort-characterization.html#cb46-72" aria-hidden="true" tabindex="-1"></a>color_set <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="st">&quot;Dx&quot;</span> <span class="ot">=</span> <span class="fu">setNames</span>(<span class="fu">okabe</span>(<span class="at">n=</span><span class="fu">length</span>(<span class="fu">unique</span>(<span class="fu">as.character</span>(pheno[,<span class="st">&quot;Dx&quot;</span>])))),</span>
<span id="cb46-73"><a href="figure-1-cohort-characterization.html#cb46-73" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">c</span>( <span class="st">&quot;CH&quot;</span>,<span class="st">&quot;MDS&quot;</span>,<span class="st">&quot;MPN&quot;</span>, <span class="st">&quot;AML&quot;</span>, <span class="st">&quot;sAML&quot;</span> ,<span class="st">&quot;tAML&quot;</span>,<span class="st">&quot;Other&quot;</span>) ),</span>
<span id="cb46-74"><a href="figure-1-cohort-characterization.html#cb46-74" aria-hidden="true" tabindex="-1"></a>                  <span class="st">&quot;Disease.Status&quot;</span> <span class="ot">=</span> <span class="fu">setNames</span>(<span class="fu">tol</span>(<span class="at">n=</span>(<span class="fu">length</span>(<span class="fu">unique</span>(<span class="fu">as.character</span>(pheno[,<span class="st">&quot;Disease.Status&quot;</span>]))))),</span>
<span id="cb46-75"><a href="figure-1-cohort-characterization.html#cb46-75" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">c</span>(<span class="st">&quot;Newly Diagnosed&quot;</span>,<span class="st">&quot;Relapse/Refractory&quot;</span>,<span class="st">&quot;Newly Transformed&quot;</span>,</span>
<span id="cb46-76"><a href="figure-1-cohort-characterization.html#cb46-76" aria-hidden="true" tabindex="-1"></a>                                      <span class="st">&quot;Persistent&quot;</span>,<span class="st">&quot;Chronic Stage&quot;</span>, <span class="st">&quot;Other&quot;</span>)))                       </span>
<span id="cb46-77"><a href="figure-1-cohort-characterization.html#cb46-77" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb46-78"><a href="figure-1-cohort-characterization.html#cb46-78" aria-hidden="true" tabindex="-1"></a><span class="co"># Format the annotations at the top of the oncoprint                                    </span></span>
<span id="cb46-79"><a href="figure-1-cohort-characterization.html#cb46-79" aria-hidden="true" tabindex="-1"></a>top_annotation <span class="ot">&lt;-</span> <span class="fu">HeatmapAnnotation</span>(<span class="at">cbar =</span> <span class="fu">anno_oncoprint_barplot</span>(),</span>
<span id="cb46-80"><a href="figure-1-cohort-characterization.html#cb46-80" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">df   =</span> pheno[,<span class="fu">c</span>(<span class="st">&quot;Dx&quot;</span>,<span class="st">&quot;Disease.Status&quot;</span>)],</span>
<span id="cb46-81"><a href="figure-1-cohort-characterization.html#cb46-81" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">col  =</span> color_set,</span>
<span id="cb46-82"><a href="figure-1-cohort-characterization.html#cb46-82" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">annotation_name_side =</span> <span class="st">&quot;left&quot;</span>)</span>
<span id="cb46-83"><a href="figure-1-cohort-characterization.html#cb46-83" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-84"><a href="figure-1-cohort-characterization.html#cb46-84" aria-hidden="true" tabindex="-1"></a><span class="co"># Indicate what should be included in the legend                  </span></span>
<span id="cb46-85"><a href="figure-1-cohort-characterization.html#cb46-85" aria-hidden="true" tabindex="-1"></a>heatmap_legend_param <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">title  =</span> <span class="st">&quot;Alternations&quot;</span>, </span>
<span id="cb46-86"><a href="figure-1-cohort-characterization.html#cb46-86" aria-hidden="true" tabindex="-1"></a>                             <span class="at">at     =</span> <span class="fu">c</span>(<span class="st">&quot;Indel&quot;</span>, <span class="st">&quot;Nonsense&quot;</span>, <span class="st">&quot;Missense&quot;</span>), </span>
<span id="cb46-87"><a href="figure-1-cohort-characterization.html#cb46-87" aria-hidden="true" tabindex="-1"></a>                             <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">&quot;Indel&quot;</span>, <span class="st">&quot;Nonsense&quot;</span>, <span class="st">&quot;Missense&quot;</span>))</span>
<span id="cb46-88"><a href="figure-1-cohort-characterization.html#cb46-88" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-89"><a href="figure-1-cohort-characterization.html#cb46-89" aria-hidden="true" tabindex="-1"></a><span class="co"># Make the oncoprint</span></span>
<span id="cb46-90"><a href="figure-1-cohort-characterization.html#cb46-90" aria-hidden="true" tabindex="-1"></a><span class="fu">oncoPrint</span>(final_mat,</span>
<span id="cb46-91"><a href="figure-1-cohort-characterization.html#cb46-91" aria-hidden="true" tabindex="-1"></a>          <span class="at">alter_fun            =</span> alter_functions, </span>
<span id="cb46-92"><a href="figure-1-cohort-characterization.html#cb46-92" aria-hidden="true" tabindex="-1"></a>          <span class="at">col                  =</span> variant_type_colors, </span>
<span id="cb46-93"><a href="figure-1-cohort-characterization.html#cb46-93" aria-hidden="true" tabindex="-1"></a>          <span class="at">top_annotation       =</span> top_annotation,</span>
<span id="cb46-94"><a href="figure-1-cohort-characterization.html#cb46-94" aria-hidden="true" tabindex="-1"></a>          <span class="at">heatmap_legend_param =</span> heatmap_legend_param)</span></code></pre></div>
<p><img src="scDNA_files/figure-html/unnamed-chunk-24-1.png" width="1152" style="display: block; margin: auto;" /></p>

</div>
<div id="sample-clonality-figure-1ce-figure-2ab" class="section level3" number="3.1.4">
<h3><span class="header-section-number">3.1.4</span> Sample clonality: Figure 1C,E Figure 2A,B</h3>
<p>Now we focused on patient samples that were included in the clonality analysis. This the filtered set o f patients that had &gt;100 cells, more than 1 mutation, and more than 1 clone following bootstrapping and estalbishing 95% confidence inervals &gt; 10 cells.</p>
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb47-1"><a href="figure-1-cohort-characterization.html#cb47-1" aria-hidden="true" tabindex="-1"></a>final_sample_summary<span class="ot">&lt;-</span><span class="fu">readRDS</span>(<span class="at">file=</span><span class="st">&quot;./data/final_sample_summary.rds&quot;</span>)</span>
<span id="cb47-2"><a href="figure-1-cohort-characterization.html#cb47-2" aria-hidden="true" tabindex="-1"></a>pheno<span class="ot">&lt;-</span><span class="fu">readRDS</span>(<span class="at">file=</span><span class="st">&quot;./data/pheno.rds&quot;</span>)</span></code></pre></div>
<p>Next we are going to build a data frame indicating the mutation status of various patients with regard to epigenetic modifiers (<em>D</em>NMT3a,<em>T</em>ET2, <em>A</em>SXL1, <em>I</em>DH1/2, DTAI) and signalling genes (FLT3, vs JAK2 vs NRAS/KRAS)</p>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb48-1"><a href="figure-1-cohort-characterization.html#cb48-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(magrittr) <span class="co"># for %&lt;&gt;% </span></span>
<span id="cb48-2"><a href="figure-1-cohort-characterization.html#cb48-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-3"><a href="figure-1-cohort-characterization.html#cb48-3" aria-hidden="true" tabindex="-1"></a><span class="co">#Tabulate presence/absence of a mutation</span></span>
<span id="cb48-4"><a href="figure-1-cohort-characterization.html#cb48-4" aria-hidden="true" tabindex="-1"></a>mutants_in_each_sample<span class="ot">&lt;-</span><span class="fu">do.call</span>(rbind,<span class="fu">lapply</span>(<span class="fu">names</span>(final_sample_summary),<span class="cf">function</span>(x){</span>
<span id="cb48-5"><a href="figure-1-cohort-characterization.html#cb48-5" aria-hidden="true" tabindex="-1"></a>  y<span class="ot">&lt;-</span><span class="fu">colnames</span>(final_sample_summary[[x]]<span class="sc">$</span>NGT)</span>
<span id="cb48-6"><a href="figure-1-cohort-characterization.html#cb48-6" aria-hidden="true" tabindex="-1"></a>  z <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb48-7"><a href="figure-1-cohort-characterization.html#cb48-7" aria-hidden="true" tabindex="-1"></a>  z<span class="sc">$</span>Sample <span class="ot">&lt;-</span> x</span>
<span id="cb48-8"><a href="figure-1-cohort-characterization.html#cb48-8" aria-hidden="true" tabindex="-1"></a>  z<span class="sc">$</span>DNMT3A <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">any</span>(<span class="fu">grepl</span>(<span class="st">&quot;DNMT3A&quot;</span>,y)),<span class="dv">1</span>,<span class="dv">0</span>)</span>
<span id="cb48-9"><a href="figure-1-cohort-characterization.html#cb48-9" aria-hidden="true" tabindex="-1"></a>  z<span class="sc">$</span>TET2 <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">any</span>(<span class="fu">grepl</span>(<span class="st">&quot;TET2&quot;</span>,y)),<span class="dv">1</span>,<span class="dv">0</span>)</span>
<span id="cb48-10"><a href="figure-1-cohort-characterization.html#cb48-10" aria-hidden="true" tabindex="-1"></a>  z<span class="sc">$</span>ASXL1 <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">any</span>(<span class="fu">grepl</span>(<span class="st">&quot;ASXL1&quot;</span>,y)),<span class="dv">1</span>,<span class="dv">0</span>)</span>
<span id="cb48-11"><a href="figure-1-cohort-characterization.html#cb48-11" aria-hidden="true" tabindex="-1"></a>  z<span class="sc">$</span>IDH <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">any</span>(<span class="fu">grepl</span>(<span class="st">&quot;IDH&quot;</span>,y)),<span class="dv">1</span>,<span class="dv">0</span>)</span>
<span id="cb48-12"><a href="figure-1-cohort-characterization.html#cb48-12" aria-hidden="true" tabindex="-1"></a>  z<span class="sc">$</span>FLT3 <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">any</span>(<span class="fu">grepl</span>(<span class="st">&quot;FLT3&quot;</span>,y)),<span class="dv">1</span>,<span class="dv">0</span>)</span>
<span id="cb48-13"><a href="figure-1-cohort-characterization.html#cb48-13" aria-hidden="true" tabindex="-1"></a>  z<span class="sc">$</span>KIT <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">any</span>(<span class="fu">grepl</span>(<span class="st">&quot;KIT&quot;</span>,y)),<span class="dv">1</span>,<span class="dv">0</span>) <span class="co"># n=1 sample, we put it in the &quot;signalling category&quot;</span></span>
<span id="cb48-14"><a href="figure-1-cohort-characterization.html#cb48-14" aria-hidden="true" tabindex="-1"></a>  z<span class="sc">$</span>RAS <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">any</span>(<span class="fu">grepl</span>(<span class="st">&quot;RAS&quot;</span>,y)),<span class="dv">1</span>,<span class="dv">0</span>)</span>
<span id="cb48-15"><a href="figure-1-cohort-characterization.html#cb48-15" aria-hidden="true" tabindex="-1"></a>  z<span class="sc">$</span>JAK2 <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">any</span>(<span class="fu">grepl</span>(<span class="st">&quot;JAK2&quot;</span>,y)),<span class="dv">1</span>,<span class="dv">0</span>)</span>
<span id="cb48-16"><a href="figure-1-cohort-characterization.html#cb48-16" aria-hidden="true" tabindex="-1"></a>  z<span class="sc">$</span>PTPN11 <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">any</span>(<span class="fu">grepl</span>(<span class="st">&quot;PTPN11&quot;</span>,y)),<span class="dv">1</span>,<span class="dv">0</span>)</span>
<span id="cb48-17"><a href="figure-1-cohort-characterization.html#cb48-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.frame</span>(<span class="fu">t</span>(<span class="fu">do.call</span>(rbind,z)))</span>
<span id="cb48-18"><a href="figure-1-cohort-characterization.html#cb48-18" aria-hidden="true" tabindex="-1"></a>}))</span>
<span id="cb48-19"><a href="figure-1-cohort-characterization.html#cb48-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-20"><a href="figure-1-cohort-characterization.html#cb48-20" aria-hidden="true" tabindex="-1"></a><span class="co"># Bin into groups based on mutations and disease type</span></span>
<span id="cb48-21"><a href="figure-1-cohort-characterization.html#cb48-21" aria-hidden="true" tabindex="-1"></a>mutants_in_each_sample<span class="sc">%&lt;&gt;%</span><span class="fu">mutate</span>(<span class="at">Group=</span><span class="fu">case_when</span>(</span>
<span id="cb48-22"><a href="figure-1-cohort-characterization.html#cb48-22" aria-hidden="true" tabindex="-1"></a>                          (TET2<span class="sc">==</span><span class="dv">1</span><span class="sc">|</span>DNMT3A<span class="sc">==</span><span class="dv">1</span><span class="sc">|</span>IDH<span class="sc">==</span><span class="dv">1</span><span class="sc">|</span>ASXL1<span class="sc">==</span><span class="dv">1</span>)<span class="sc">&amp;</span>(RAS<span class="sc">==</span><span class="dv">0</span><span class="sc">&amp;</span>FLT3<span class="sc">==</span><span class="dv">0</span>)<span class="sc">~</span><span class="st">&#39;DTAI&#39;</span>,</span>
<span id="cb48-23"><a href="figure-1-cohort-characterization.html#cb48-23" aria-hidden="true" tabindex="-1"></a>                          (TET2<span class="sc">==</span><span class="dv">1</span><span class="sc">|</span>DNMT3A<span class="sc">==</span><span class="dv">1</span><span class="sc">|</span>IDH<span class="sc">==</span><span class="dv">1</span><span class="sc">|</span>ASXL1<span class="sc">==</span><span class="dv">1</span>)<span class="sc">&amp;</span>((RAS<span class="sc">==</span><span class="dv">1</span><span class="sc">&amp;</span>FLT3<span class="sc">==</span><span class="dv">0</span>)<span class="sc">|</span></span>
<span id="cb48-24"><a href="figure-1-cohort-characterization.html#cb48-24" aria-hidden="true" tabindex="-1"></a>                                                                   (PTPN11<span class="sc">==</span><span class="dv">1</span><span class="sc">&amp;</span>FLT3<span class="sc">==</span><span class="dv">0</span>))<span class="sc">~</span><span class="st">&#39;DTAI-RAS&#39;</span>,</span>
<span id="cb48-25"><a href="figure-1-cohort-characterization.html#cb48-25" aria-hidden="true" tabindex="-1"></a>                          (TET2<span class="sc">==</span><span class="dv">1</span><span class="sc">|</span>DNMT3A<span class="sc">==</span><span class="dv">1</span><span class="sc">|</span>IDH<span class="sc">==</span><span class="dv">1</span><span class="sc">|</span>ASXL1<span class="sc">==</span><span class="dv">1</span>)<span class="sc">&amp;</span>(RAS<span class="sc">==</span><span class="dv">0</span><span class="sc">&amp;</span>FLT3<span class="sc">==</span><span class="dv">1</span>)<span class="sc">~</span><span class="st">&#39;DTAI-FLT3&#39;</span>,</span>
<span id="cb48-26"><a href="figure-1-cohort-characterization.html#cb48-26" aria-hidden="true" tabindex="-1"></a>                          (TET2<span class="sc">==</span><span class="dv">1</span><span class="sc">|</span>DNMT3A<span class="sc">==</span><span class="dv">1</span><span class="sc">|</span>IDH<span class="sc">==</span><span class="dv">1</span><span class="sc">|</span>ASXL1<span class="sc">==</span><span class="dv">1</span>)<span class="sc">&amp;</span>((RAS<span class="sc">==</span><span class="dv">1</span><span class="sc">&amp;</span>FLT3<span class="sc">==</span><span class="dv">1</span>)<span class="sc">|</span></span>
<span id="cb48-27"><a href="figure-1-cohort-characterization.html#cb48-27" aria-hidden="true" tabindex="-1"></a>                                                               (PTPN11<span class="sc">==</span><span class="dv">1</span><span class="sc">&amp;</span>FLT3<span class="sc">==</span><span class="dv">1</span>))<span class="sc">~</span><span class="st">&#39;DTAI-FLT3-RAS&#39;</span>,</span>
<span id="cb48-28"><a href="figure-1-cohort-characterization.html#cb48-28" aria-hidden="true" tabindex="-1"></a>                          (TET2<span class="sc">==</span><span class="dv">0</span><span class="sc">&amp;</span>DNMT3A<span class="sc">==</span><span class="dv">0</span><span class="sc">&amp;</span>IDH<span class="sc">==</span><span class="dv">0</span><span class="sc">&amp;</span>ASXL1<span class="sc">==</span><span class="dv">0</span>)<span class="sc">&amp;</span>(RAS<span class="sc">==</span><span class="dv">1</span><span class="sc">|</span>FLT3<span class="sc">==</span><span class="dv">1</span><span class="sc">|</span>JAK2<span class="sc">==</span><span class="dv">1</span><span class="sc">|</span>KIT<span class="sc">==</span><span class="dv">1</span>)<span class="sc">~</span><span class="st">&#39;Signaling&#39;</span>))<span class="sc">%&gt;%</span></span>
<span id="cb48-29"><a href="figure-1-cohort-characterization.html#cb48-29" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">left_join</span>(pheno,<span class="at">by=</span><span class="st">&quot;Sample&quot;</span>)<span class="sc">%&gt;%</span></span>
<span id="cb48-30"><a href="figure-1-cohort-characterization.html#cb48-30" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">mutate</span>(<span class="at">Final_group=</span><span class="fu">case_when</span>(</span>
<span id="cb48-31"><a href="figure-1-cohort-characterization.html#cb48-31" aria-hidden="true" tabindex="-1"></a>                                          <span class="fu">grepl</span>(<span class="st">&quot;AML|Other&quot;</span>,Dx)<span class="sc">~</span>Group,</span>
<span id="cb48-32"><a href="figure-1-cohort-characterization.html#cb48-32" aria-hidden="true" tabindex="-1"></a>                                          <span class="sc">!</span><span class="fu">grepl</span>(<span class="st">&quot;AML|Other&quot;</span>,Dx)<span class="sc">~</span>Dx</span>
<span id="cb48-33"><a href="figure-1-cohort-characterization.html#cb48-33" aria-hidden="true" tabindex="-1"></a>                                        ))</span>
<span id="cb48-34"><a href="figure-1-cohort-characterization.html#cb48-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-35"><a href="figure-1-cohort-characterization.html#cb48-35" aria-hidden="true" tabindex="-1"></a><span class="co"># Order the groups to match how we have them in the paper</span></span>
<span id="cb48-36"><a href="figure-1-cohort-characterization.html#cb48-36" aria-hidden="true" tabindex="-1"></a>mutants_in_each_sample<span class="sc">$</span>Final_group <span class="ot">&lt;-</span> <span class="fu">factor</span>(mutants_in_each_sample<span class="sc">$</span>Final_group,</span>
<span id="cb48-37"><a href="figure-1-cohort-characterization.html#cb48-37" aria-hidden="true" tabindex="-1"></a>                                              <span class="at">levels=</span><span class="fu">c</span>(<span class="st">&quot;CH&quot;</span>,<span class="st">&quot;MPN&quot;</span>,<span class="st">&quot;Signaling&quot;</span>,<span class="st">&quot;DTAI&quot;</span>,</span>
<span id="cb48-38"><a href="figure-1-cohort-characterization.html#cb48-38" aria-hidden="true" tabindex="-1"></a>                                                       <span class="st">&quot;DTAI-RAS&quot;</span>,<span class="st">&quot;DTAI-FLT3&quot;</span>,<span class="st">&quot;DTAI-FLT3-RAS&quot;</span>))</span></code></pre></div>
<p>Next we want to calculate a few metrics for each patient sample.</p>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb49-1"><a href="figure-1-cohort-characterization.html#cb49-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(vegan) </span>
<span id="cb49-2"><a href="figure-1-cohort-characterization.html#cb49-2" aria-hidden="true" tabindex="-1"></a>clonal_level_info<span class="ot">&lt;-</span><span class="fu">data.frame</span>(<span class="fu">do.call</span>(rbind,<span class="fu">lapply</span>(<span class="fu">names</span>(final_sample_summary),<span class="cf">function</span>(y){</span>
<span id="cb49-3"><a href="figure-1-cohort-characterization.html#cb49-3" aria-hidden="true" tabindex="-1"></a>  x <span class="ot">&lt;-</span> final_sample_summary[[y]]<span class="sc">$</span>Clones</span>
<span id="cb49-4"><a href="figure-1-cohort-characterization.html#cb49-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.frame</span>(<span class="st">&quot;Sample&quot;</span><span class="ot">=</span>y,</span>
<span id="cb49-5"><a href="figure-1-cohort-characterization.html#cb49-5" aria-hidden="true" tabindex="-1"></a>             <span class="st">&quot;Shannon&quot;</span><span class="ot">=</span>vegan<span class="sc">::</span><span class="fu">diversity</span>(x[,<span class="dv">1</span>],<span class="at">index=</span><span class="st">&quot;shannon&quot;</span>),</span>
<span id="cb49-6"><a href="figure-1-cohort-characterization.html#cb49-6" aria-hidden="true" tabindex="-1"></a>             <span class="st">&quot;Number_of_clones&quot;</span><span class="ot">=</span><span class="fu">length</span>(x[,<span class="dv">1</span>]),</span>
<span id="cb49-7"><a href="figure-1-cohort-characterization.html#cb49-7" aria-hidden="true" tabindex="-1"></a>             <span class="st">&quot;Number_of_mutations&quot;</span><span class="ot">=</span><span class="fu">ncol</span>(final_sample_summary[[y]]<span class="sc">$</span>NGT),</span>
<span id="cb49-8"><a href="figure-1-cohort-characterization.html#cb49-8" aria-hidden="true" tabindex="-1"></a>             <span class="st">&quot;Number_of_mutations_in_dominant_clone&quot;</span><span class="ot">=</span><span class="fu">sum</span>(<span class="fu">as.numeric</span>(<span class="fu">do.call</span>(rbind,</span>
<span id="cb49-9"><a href="figure-1-cohort-characterization.html#cb49-9" aria-hidden="true" tabindex="-1"></a>                                           <span class="fu">strsplit</span>(<span class="fu">as.character</span>(x[<span class="fu">nrow</span>(x),<span class="dv">2</span>]),<span class="at">split=</span><span class="st">&quot;_&quot;</span>)))),</span>
<span id="cb49-10"><a href="figure-1-cohort-characterization.html#cb49-10" aria-hidden="true" tabindex="-1"></a>             <span class="st">&quot;Dominant_clone_size&quot;</span><span class="ot">=</span><span class="fu">max</span>(x[,<span class="dv">1</span>]<span class="sc">/</span><span class="fu">sum</span>(x[,<span class="dv">1</span>]))) <span class="co">#,</span></span>
<span id="cb49-11"><a href="figure-1-cohort-characterization.html#cb49-11" aria-hidden="true" tabindex="-1"></a>           })))</span></code></pre></div>
<p>Now we’ll merge the data frames together and plot some of the data found in Figure 1 and Figure 2</p>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb50-1"><a href="figure-1-cohort-characterization.html#cb50-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine the data frame</span></span>
<span id="cb50-2"><a href="figure-1-cohort-characterization.html#cb50-2" aria-hidden="true" tabindex="-1"></a>test<span class="ot">&lt;-</span>mutants_in_each_sample<span class="sc">%&gt;%</span><span class="fu">inner_join</span>(clonal_level_info)</span>
<span id="cb50-3"><a href="figure-1-cohort-characterization.html#cb50-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-4"><a href="figure-1-cohort-characterization.html#cb50-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Number of mutations</span></span>
<span id="cb50-5"><a href="figure-1-cohort-characterization.html#cb50-5" aria-hidden="true" tabindex="-1"></a>gg_number_of_mutations<span class="ot">&lt;-</span><span class="fu">ggplot</span>(test<span class="sc">%&gt;%</span><span class="fu">group_by</span>(Final_group)<span class="sc">%&gt;%</span></span>
<span id="cb50-6"><a href="figure-1-cohort-characterization.html#cb50-6" aria-hidden="true" tabindex="-1"></a>                                  <span class="fu">summarise</span>(<span class="at">mean=</span><span class="fu">mean</span>(Number_of_mutations),</span>
<span id="cb50-7"><a href="figure-1-cohort-characterization.html#cb50-7" aria-hidden="true" tabindex="-1"></a>                                            <span class="at">sd =</span> <span class="fu">sd</span>(Number_of_mutations),</span>
<span id="cb50-8"><a href="figure-1-cohort-characterization.html#cb50-8" aria-hidden="true" tabindex="-1"></a>                                            <span class="at">sem =</span> <span class="fu">sd</span>(Number_of_mutations)<span class="sc">/</span></span>
<span id="cb50-9"><a href="figure-1-cohort-characterization.html#cb50-9" aria-hidden="true" tabindex="-1"></a>                                                      <span class="fu">sqrt</span>(<span class="fu">length</span>(Number_of_mutations))),</span>
<span id="cb50-10"><a href="figure-1-cohort-characterization.html#cb50-10" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">aes</span>(<span class="at">x=</span>Final_group,<span class="at">y=</span>mean,<span class="at">fill=</span>Final_group))<span class="sc">+</span></span>
<span id="cb50-11"><a href="figure-1-cohort-characterization.html#cb50-11" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">&quot;identity&quot;</span>,<span class="at">color=</span><span class="st">&quot;black&quot;</span>)<span class="sc">+</span></span>
<span id="cb50-12"><a href="figure-1-cohort-characterization.html#cb50-12" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">geom_errorbar</span>(<span class="fu">aes</span>(<span class="at">ymin =</span> mean<span class="sc">-</span>sem, <span class="at">ymax =</span> mean<span class="sc">+</span>sem),<span class="at">width=</span><span class="fl">0.5</span>,<span class="at">lwd=</span><span class="fl">0.5</span>)<span class="sc">+</span></span>
<span id="cb50-13"><a href="figure-1-cohort-characterization.html#cb50-13" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">theme_classic</span>(<span class="at">base_size =</span> <span class="dv">8</span>)<span class="sc">+</span></span>
<span id="cb50-14"><a href="figure-1-cohort-characterization.html#cb50-14" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">ylab</span>(<span class="st">&quot;Number of mutations&quot;</span>)<span class="sc">+</span><span class="fu">xlab</span>(<span class="st">&quot;&quot;</span>)<span class="sc">+</span><span class="fu">ggtitle</span>(<span class="st">&quot;&quot;</span>)<span class="sc">+</span></span>
<span id="cb50-15"><a href="figure-1-cohort-characterization.html#cb50-15" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">scale_y_continuous</span>(<span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">9</span>), <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb50-16"><a href="figure-1-cohort-characterization.html#cb50-16" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle=</span><span class="dv">30</span>,<span class="at">hjust=</span><span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb50-17"><a href="figure-1-cohort-characterization.html#cb50-17" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">scale_fill_brewer</span>(<span class="at">type=</span><span class="st">&quot;seq&quot;</span>,<span class="at">palette =</span> <span class="st">&quot;Reds&quot;</span>,<span class="at">aesthetics =</span> <span class="st">&quot;fill&quot;</span>,<span class="at">guide=</span><span class="cn">FALSE</span>)</span>
<span id="cb50-18"><a href="figure-1-cohort-characterization.html#cb50-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-19"><a href="figure-1-cohort-characterization.html#cb50-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Number of clones</span></span>
<span id="cb50-20"><a href="figure-1-cohort-characterization.html#cb50-20" aria-hidden="true" tabindex="-1"></a>gg_number_of_clones<span class="ot">&lt;-</span><span class="fu">ggplot</span>(test,<span class="fu">aes</span>(<span class="at">y=</span>Number_of_clones,<span class="at">x=</span>Final_group,<span class="at">fill=</span>Final_group))<span class="sc">+</span></span>
<span id="cb50-21"><a href="figure-1-cohort-characterization.html#cb50-21" aria-hidden="true" tabindex="-1"></a>                                  <span class="fu">geom_boxplot</span>(<span class="at">outlier.shape =</span> <span class="cn">NA</span>)<span class="sc">+</span>  </span>
<span id="cb50-22"><a href="figure-1-cohort-characterization.html#cb50-22" aria-hidden="true" tabindex="-1"></a>                                  <span class="fu">geom_jitter</span>(<span class="at">width =</span> <span class="fl">0.1</span>,<span class="at">size=</span><span class="fl">0.5</span>)<span class="sc">+</span></span>
<span id="cb50-23"><a href="figure-1-cohort-characterization.html#cb50-23" aria-hidden="true" tabindex="-1"></a>                                  <span class="fu">theme_classic</span>(<span class="at">base_size =</span> <span class="dv">8</span>)<span class="sc">+</span></span>
<span id="cb50-24"><a href="figure-1-cohort-characterization.html#cb50-24" aria-hidden="true" tabindex="-1"></a>                                  <span class="fu">ylab</span>(<span class="st">&quot;Number of clones&quot;</span>)<span class="sc">+</span></span>
<span id="cb50-25"><a href="figure-1-cohort-characterization.html#cb50-25" aria-hidden="true" tabindex="-1"></a>                                  <span class="fu">xlab</span>(<span class="st">&quot;&quot;</span>)<span class="sc">+</span></span>
<span id="cb50-26"><a href="figure-1-cohort-characterization.html#cb50-26" aria-hidden="true" tabindex="-1"></a>                                  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle=</span><span class="dv">30</span>,<span class="at">hjust=</span><span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb50-27"><a href="figure-1-cohort-characterization.html#cb50-27" aria-hidden="true" tabindex="-1"></a>                                  <span class="fu">scale_fill_brewer</span>(<span class="at">type=</span><span class="st">&quot;seq&quot;</span>,<span class="at">palette =</span> <span class="st">&quot;Reds&quot;</span>,<span class="at">aesthetics =</span> <span class="st">&quot;fill&quot;</span>,<span class="at">guide=</span><span class="cn">FALSE</span>)</span>
<span id="cb50-28"><a href="figure-1-cohort-characterization.html#cb50-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-29"><a href="figure-1-cohort-characterization.html#cb50-29" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_grid</span>(gg_number_of_mutations,gg_number_of_clones,<span class="at">ncol=</span><span class="dv">2</span>,<span class="at">align=</span><span class="st">&quot;hv&quot;</span>,<span class="at">axis=</span><span class="st">&quot;ltrb&quot;</span>,<span class="at">labels=</span><span class="fu">c</span>(<span class="st">&quot;C&quot;</span>,<span class="st">&quot;E&quot;</span>))</span></code></pre></div>
<div class="figure" style="text-align: center"><span id="fig:unnamed-chunk-28"></span>
<img src="scDNA_files/figure-html/unnamed-chunk-28-1.png" alt="Miles et al: Figure 1C,E" width="700.8" />
<p class="caption">
Figure 3.2: Miles et al: Figure 1C,E
</p>
</div>
<p>Compute statistics for the different group comparisons. We used a Benjamini &amp; Hochberg FDR for multiple test correction with a significance cutoff of 0.1.</p>
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb51-1"><a href="figure-1-cohort-characterization.html#cb51-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(reshape2) <span class="co">#for melt, I need to come up with a better way to do this, if anyone has ideas let me know!</span></span>
<span id="cb51-2"><a href="figure-1-cohort-characterization.html#cb51-2" aria-hidden="true" tabindex="-1"></a>pvalues_Number_of_clones<span class="ot">&lt;-</span>test<span class="sc">%&gt;%</span>{<span class="fu">melt</span>(<span class="fu">pairwise.t.test</span>(.<span class="sc">$</span>Number_of_clones,<span class="at">g=</span>.<span class="sc">$</span>Final_group,</span>
<span id="cb51-3"><a href="figure-1-cohort-characterization.html#cb51-3" aria-hidden="true" tabindex="-1"></a>                                                     <span class="at">data=</span>.,<span class="at">p.adjust.method=</span><span class="st">&quot;fdr&quot;</span>)<span class="sc">$</span>p.value)}<span class="sc">%&gt;%</span></span>
<span id="cb51-4"><a href="figure-1-cohort-characterization.html#cb51-4" aria-hidden="true" tabindex="-1"></a>                                     <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(value))<span class="sc">%&gt;%</span><span class="fu">filter</span>(value<span class="sc">&lt;</span><span class="fl">0.1</span>)</span>
<span id="cb51-5"><a href="figure-1-cohort-characterization.html#cb51-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-6"><a href="figure-1-cohort-characterization.html#cb51-6" aria-hidden="true" tabindex="-1"></a>pvalues_Number_of_mutations<span class="ot">&lt;-</span>test<span class="sc">%&gt;%</span>{<span class="fu">melt</span>(<span class="fu">pairwise.t.test</span>(.<span class="sc">$</span>Number_of_mutations,<span class="at">g=</span>.<span class="sc">$</span>Final_group,</span>
<span id="cb51-7"><a href="figure-1-cohort-characterization.html#cb51-7" aria-hidden="true" tabindex="-1"></a>                                                     <span class="at">data=</span>.,<span class="at">p.adjust.method=</span><span class="st">&quot;fdr&quot;</span>)<span class="sc">$</span>p.value)}<span class="sc">%&gt;%</span></span>
<span id="cb51-8"><a href="figure-1-cohort-characterization.html#cb51-8" aria-hidden="true" tabindex="-1"></a>                                     <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(value))<span class="sc">%&gt;%</span><span class="fu">filter</span>(value<span class="sc">&lt;</span><span class="fl">0.1</span>)</span></code></pre></div>
<table class="table" style="width: auto !important; float: left; margin-right: 10px;">
<caption>
<span id="tab:unnamed-chunk-30">Table 3.1: </span>Number of Clones
</caption>
<thead>
<tr>
<th style="text-align:left;">
Group 1
</th>
<th style="text-align:left;">
Group 2
</th>
<th style="text-align:right;">
FDR
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
DTAI-RAS
</td>
<td style="text-align:left;">
CH
</td>
<td style="text-align:right;">
0.0466140
</td>
</tr>
<tr>
<td style="text-align:left;">
DTAI-FLT3
</td>
<td style="text-align:left;">
CH
</td>
<td style="text-align:right;">
0.0001112
</td>
</tr>
<tr>
<td style="text-align:left;">
DTAI-FLT3-RAS
</td>
<td style="text-align:left;">
CH
</td>
<td style="text-align:right;">
0.0245701
</td>
</tr>
<tr>
<td style="text-align:left;">
DTAI-RAS
</td>
<td style="text-align:left;">
MPN
</td>
<td style="text-align:right;">
0.0899428
</td>
</tr>
<tr>
<td style="text-align:left;">
DTAI-FLT3
</td>
<td style="text-align:left;">
MPN
</td>
<td style="text-align:right;">
0.0001654
</td>
</tr>
<tr>
<td style="text-align:left;">
DTAI-FLT3-RAS
</td>
<td style="text-align:left;">
MPN
</td>
<td style="text-align:right;">
0.0459698
</td>
</tr>
<tr>
<td style="text-align:left;">
DTAI-FLT3
</td>
<td style="text-align:left;">
Signaling
</td>
<td style="text-align:right;">
0.0157077
</td>
</tr>
<tr>
<td style="text-align:left;">
DTAI-FLT3
</td>
<td style="text-align:left;">
DTAI
</td>
<td style="text-align:right;">
0.0001112
</td>
</tr>
<tr>
<td style="text-align:left;">
DTAI-FLT3-RAS
</td>
<td style="text-align:left;">
DTAI
</td>
<td style="text-align:right;">
0.0637393
</td>
</tr>
<tr>
<td style="text-align:left;">
DTAI-FLT3
</td>
<td style="text-align:left;">
DTAI-RAS
</td>
<td style="text-align:right;">
0.0105738
</td>
</tr>
</tbody>
</table>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:unnamed-chunk-30">Table 3.1: </span>Number of Mutations
</caption>
<thead>
<tr>
<th style="text-align:left;">
Group 1
</th>
<th style="text-align:left;">
Group 2
</th>
<th style="text-align:right;">
FDR
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Signaling
</td>
<td style="text-align:left;">
CH
</td>
<td style="text-align:right;">
0.0171093
</td>
</tr>
<tr>
<td style="text-align:left;">
DTAI-RAS
</td>
<td style="text-align:left;">
CH
</td>
<td style="text-align:right;">
0.0763530
</td>
</tr>
<tr>
<td style="text-align:left;">
DTAI-FLT3
</td>
<td style="text-align:left;">
CH
</td>
<td style="text-align:right;">
0.0000064
</td>
</tr>
<tr>
<td style="text-align:left;">
DTAI-FLT3-RAS
</td>
<td style="text-align:left;">
CH
</td>
<td style="text-align:right;">
0.0040371
</td>
</tr>
<tr>
<td style="text-align:left;">
Signaling
</td>
<td style="text-align:left;">
MPN
</td>
<td style="text-align:right;">
0.0729728
</td>
</tr>
<tr>
<td style="text-align:left;">
DTAI-FLT3
</td>
<td style="text-align:left;">
MPN
</td>
<td style="text-align:right;">
0.0000448
</td>
</tr>
<tr>
<td style="text-align:left;">
DTAI-FLT3-RAS
</td>
<td style="text-align:left;">
MPN
</td>
<td style="text-align:right;">
0.0171093
</td>
</tr>
<tr>
<td style="text-align:left;">
DTAI
</td>
<td style="text-align:left;">
Signaling
</td>
<td style="text-align:right;">
0.0017335
</td>
</tr>
<tr>
<td style="text-align:left;">
DTAI-FLT3
</td>
<td style="text-align:left;">
Signaling
</td>
<td style="text-align:right;">
0.0249373
</td>
</tr>
<tr>
<td style="text-align:left;">
DTAI-RAS
</td>
<td style="text-align:left;">
DTAI
</td>
<td style="text-align:right;">
0.0058893
</td>
</tr>
<tr>
<td style="text-align:left;">
DTAI-FLT3
</td>
<td style="text-align:left;">
DTAI
</td>
<td style="text-align:right;">
0.0000000
</td>
</tr>
<tr>
<td style="text-align:left;">
DTAI-FLT3-RAS
</td>
<td style="text-align:left;">
DTAI
</td>
<td style="text-align:right;">
0.0003139
</td>
</tr>
<tr>
<td style="text-align:left;">
DTAI-FLT3
</td>
<td style="text-align:left;">
DTAI-RAS
</td>
<td style="text-align:right;">
0.0000812
</td>
</tr>
<tr>
<td style="text-align:left;">
DTAI-FLT3-RAS
</td>
<td style="text-align:left;">
DTAI-RAS
</td>
<td style="text-align:right;">
0.0658341
</td>
</tr>
</tbody>
</table>
<p>Now for Figure 2:</p>
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb52-1"><a href="figure-1-cohort-characterization.html#cb52-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Shannon diversity index</span></span>
<span id="cb52-2"><a href="figure-1-cohort-characterization.html#cb52-2" aria-hidden="true" tabindex="-1"></a>gg_shannon<span class="ot">&lt;-</span><span class="fu">ggplot</span>(test,<span class="fu">aes</span>(<span class="at">y=</span>Shannon,<span class="at">x=</span>Final_group,<span class="at">fill=</span>Final_group))<span class="sc">+</span></span>
<span id="cb52-3"><a href="figure-1-cohort-characterization.html#cb52-3" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">geom_boxplot</span>(<span class="at">outlier.shape =</span> <span class="cn">NA</span>)<span class="sc">+</span>  </span>
<span id="cb52-4"><a href="figure-1-cohort-characterization.html#cb52-4" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">geom_jitter</span>(<span class="at">width =</span> <span class="fl">0.1</span>,<span class="at">size=</span><span class="fl">0.5</span>)<span class="sc">+</span></span>
<span id="cb52-5"><a href="figure-1-cohort-characterization.html#cb52-5" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">theme_classic</span>(<span class="at">base_size =</span> <span class="dv">8</span>)<span class="sc">+</span></span>
<span id="cb52-6"><a href="figure-1-cohort-characterization.html#cb52-6" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">ylab</span>(<span class="st">&quot;Shannon diveristy index&quot;</span>)<span class="sc">+</span></span>
<span id="cb52-7"><a href="figure-1-cohort-characterization.html#cb52-7" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">xlab</span>(<span class="st">&quot;&quot;</span>)<span class="sc">+</span></span>
<span id="cb52-8"><a href="figure-1-cohort-characterization.html#cb52-8" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle=</span><span class="dv">30</span>,<span class="at">hjust=</span><span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb52-9"><a href="figure-1-cohort-characterization.html#cb52-9" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">scale_fill_brewer</span>(<span class="at">type=</span><span class="st">&quot;seq&quot;</span>,<span class="at">palette =</span> <span class="st">&quot;Reds&quot;</span>,<span class="at">aesthetics =</span> <span class="st">&quot;fill&quot;</span>,<span class="at">guide=</span><span class="cn">FALSE</span>)</span>
<span id="cb52-10"><a href="figure-1-cohort-characterization.html#cb52-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-11"><a href="figure-1-cohort-characterization.html#cb52-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Number of mutations in each cohort</span></span>
<span id="cb52-12"><a href="figure-1-cohort-characterization.html#cb52-12" aria-hidden="true" tabindex="-1"></a>gg_Number_of_mutations_in_Dclone<span class="ot">&lt;-</span><span class="fu">ggplot</span>(test<span class="sc">%&gt;%</span><span class="fu">group_by</span>(Final_group)<span class="sc">%&gt;%</span></span>
<span id="cb52-13"><a href="figure-1-cohort-characterization.html#cb52-13" aria-hidden="true" tabindex="-1"></a>                                       <span class="fu">summarise</span>(<span class="at">mean=</span><span class="fu">mean</span>(Number_of_mutations_in_dominant_clone),</span>
<span id="cb52-14"><a href="figure-1-cohort-characterization.html#cb52-14" aria-hidden="true" tabindex="-1"></a>                                                 <span class="at">sd =</span> <span class="fu">sd</span>(Number_of_mutations_in_dominant_clone),</span>
<span id="cb52-15"><a href="figure-1-cohort-characterization.html#cb52-15" aria-hidden="true" tabindex="-1"></a>                                                 <span class="at">sem =</span> <span class="fu">sd</span>(Number_of_mutations_in_dominant_clone)<span class="sc">/</span></span>
<span id="cb52-16"><a href="figure-1-cohort-characterization.html#cb52-16" aria-hidden="true" tabindex="-1"></a>                                                       <span class="fu">sqrt</span>(<span class="fu">length</span>(Number_of_mutations_in_dominant_clone))),</span>
<span id="cb52-17"><a href="figure-1-cohort-characterization.html#cb52-17" aria-hidden="true" tabindex="-1"></a>                                       <span class="fu">aes</span>(<span class="at">x=</span>Final_group,<span class="at">y=</span>mean,<span class="at">fill=</span>Final_group))<span class="sc">+</span></span>
<span id="cb52-18"><a href="figure-1-cohort-characterization.html#cb52-18" aria-hidden="true" tabindex="-1"></a>                                          <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">&quot;identity&quot;</span>,<span class="at">color=</span><span class="st">&quot;black&quot;</span>)<span class="sc">+</span></span>
<span id="cb52-19"><a href="figure-1-cohort-characterization.html#cb52-19" aria-hidden="true" tabindex="-1"></a>                                          <span class="fu">geom_errorbar</span>(<span class="fu">aes</span>(<span class="at">ymin =</span> mean<span class="sc">-</span>sem, <span class="at">ymax =</span> mean<span class="sc">+</span>sem),<span class="at">width=</span><span class="fl">0.5</span>,<span class="at">lwd=</span><span class="fl">0.5</span>)<span class="sc">+</span></span>
<span id="cb52-20"><a href="figure-1-cohort-characterization.html#cb52-20" aria-hidden="true" tabindex="-1"></a>                                          <span class="fu">theme_classic</span>(<span class="at">base_size =</span> <span class="dv">8</span>)<span class="sc">+</span></span>
<span id="cb52-21"><a href="figure-1-cohort-characterization.html#cb52-21" aria-hidden="true" tabindex="-1"></a>                                          <span class="fu">ylab</span>(<span class="st">&quot;Number of mutations </span><span class="sc">\n</span><span class="st"> in dominant clone&quot;</span>)<span class="sc">+</span><span class="fu">xlab</span>(<span class="st">&quot;&quot;</span>)<span class="sc">+</span><span class="fu">ggtitle</span>(<span class="st">&quot;&quot;</span>)<span class="sc">+</span></span>
<span id="cb52-22"><a href="figure-1-cohort-characterization.html#cb52-22" aria-hidden="true" tabindex="-1"></a>                                          <span class="fu">scale_y_continuous</span>(<span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">4.5</span>), <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb52-23"><a href="figure-1-cohort-characterization.html#cb52-23" aria-hidden="true" tabindex="-1"></a>                                          <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle=</span><span class="dv">30</span>,<span class="at">hjust=</span><span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb52-24"><a href="figure-1-cohort-characterization.html#cb52-24" aria-hidden="true" tabindex="-1"></a>                                          <span class="fu">scale_fill_brewer</span>(<span class="at">type=</span><span class="st">&quot;seq&quot;</span>,<span class="at">palette =</span> <span class="st">&quot;Reds&quot;</span>,</span>
<span id="cb52-25"><a href="figure-1-cohort-characterization.html#cb52-25" aria-hidden="true" tabindex="-1"></a>                                                            <span class="at">aesthetics =</span> <span class="st">&quot;fill&quot;</span>,<span class="at">guide=</span><span class="cn">FALSE</span>)</span>
<span id="cb52-26"><a href="figure-1-cohort-characterization.html#cb52-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-27"><a href="figure-1-cohort-characterization.html#cb52-27" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_grid</span>(gg_shannon,gg_Number_of_mutations_in_Dclone,<span class="at">ncol=</span><span class="dv">2</span>,<span class="at">align=</span><span class="st">&quot;hv&quot;</span>,<span class="at">axis=</span><span class="st">&quot;ltrb&quot;</span>,<span class="at">labels=</span><span class="fu">c</span>(<span class="st">&quot;A&quot;</span>,<span class="st">&quot;B&quot;</span>))</span></code></pre></div>
<div class="figure" style="text-align: center"><span id="fig:unnamed-chunk-31"></span>
<img src="scDNA_files/figure-html/unnamed-chunk-31-1.png" alt="Miles et al: Figure 2A-B" width="700.8" />
<p class="caption">
Figure 3.3: Miles et al: Figure 2A-B
</p>
</div>
<div class="sourceCode" id="cb53"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb53-1"><a href="figure-1-cohort-characterization.html#cb53-1" aria-hidden="true" tabindex="-1"></a>pvalues_Shannon<span class="ot">&lt;-</span>test<span class="sc">%&gt;%</span>{<span class="fu">melt</span>(<span class="fu">pairwise.t.test</span>(.<span class="sc">$</span>Shannon,<span class="at">g=</span>.<span class="sc">$</span>Final_group,</span>
<span id="cb53-2"><a href="figure-1-cohort-characterization.html#cb53-2" aria-hidden="true" tabindex="-1"></a>                                                        <span class="at">data=</span>.,<span class="at">p.adjust.method=</span><span class="st">&quot;fdr&quot;</span>)<span class="sc">$</span>p.value)}<span class="sc">%&gt;%</span></span>
<span id="cb53-3"><a href="figure-1-cohort-characterization.html#cb53-3" aria-hidden="true" tabindex="-1"></a>                                              <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(value))<span class="sc">%&gt;%</span><span class="fu">filter</span>(value<span class="sc">&lt;</span><span class="fl">0.1</span>)</span>
<span id="cb53-4"><a href="figure-1-cohort-characterization.html#cb53-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb53-5"><a href="figure-1-cohort-characterization.html#cb53-5" aria-hidden="true" tabindex="-1"></a>pvalues_Number_of_mutations_in_dominant_clone<span class="ot">&lt;-</span>test<span class="sc">%&gt;%</span>{<span class="fu">melt</span>(<span class="fu">pairwise.t.test</span>(</span>
<span id="cb53-6"><a href="figure-1-cohort-characterization.html#cb53-6" aria-hidden="true" tabindex="-1"></a>                                                        .<span class="sc">$</span>Number_of_mutations_in_dominant_clone,</span>
<span id="cb53-7"><a href="figure-1-cohort-characterization.html#cb53-7" aria-hidden="true" tabindex="-1"></a>                                                        <span class="at">g=</span>.<span class="sc">$</span>Final_group,</span>
<span id="cb53-8"><a href="figure-1-cohort-characterization.html#cb53-8" aria-hidden="true" tabindex="-1"></a>                                                        <span class="at">data=</span>.,<span class="at">p.adjust.method=</span><span class="st">&quot;fdr&quot;</span>)<span class="sc">$</span>p.value)}<span class="sc">%&gt;%</span></span>
<span id="cb53-9"><a href="figure-1-cohort-characterization.html#cb53-9" aria-hidden="true" tabindex="-1"></a>                                              <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(value))<span class="sc">%&gt;%</span><span class="fu">filter</span>(value<span class="sc">&lt;</span><span class="fl">0.1</span>)</span></code></pre></div>
<table class="table" style="width: auto !important; float: left; margin-right: 10px;">
<caption>
<span id="tab:unnamed-chunk-32">Table 3.2: </span>Shannon diversity index
</caption>
<thead>
<tr>
<th style="text-align:left;">
Group 1
</th>
<th style="text-align:left;">
Group 2
</th>
<th style="text-align:right;">
FDR
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Signaling
</td>
<td style="text-align:left;">
CH
</td>
<td style="text-align:right;">
0.0675230
</td>
</tr>
<tr>
<td style="text-align:left;">
DTAI-RAS
</td>
<td style="text-align:left;">
CH
</td>
<td style="text-align:right;">
0.0048421
</td>
</tr>
<tr>
<td style="text-align:left;">
DTAI-FLT3
</td>
<td style="text-align:left;">
CH
</td>
<td style="text-align:right;">
0.0000925
</td>
</tr>
<tr>
<td style="text-align:left;">
DTAI-FLT3-RAS
</td>
<td style="text-align:left;">
CH
</td>
<td style="text-align:right;">
0.0082978
</td>
</tr>
<tr>
<td style="text-align:left;">
DTAI-RAS
</td>
<td style="text-align:left;">
MPN
</td>
<td style="text-align:right;">
0.0327734
</td>
</tr>
<tr>
<td style="text-align:left;">
DTAI-FLT3
</td>
<td style="text-align:left;">
MPN
</td>
<td style="text-align:right;">
0.0006562
</td>
</tr>
<tr>
<td style="text-align:left;">
DTAI-FLT3-RAS
</td>
<td style="text-align:left;">
MPN
</td>
<td style="text-align:right;">
0.0342246
</td>
</tr>
<tr>
<td style="text-align:left;">
DTAI-FLT3
</td>
<td style="text-align:left;">
Signaling
</td>
<td style="text-align:right;">
0.0266874
</td>
</tr>
<tr>
<td style="text-align:left;">
DTAI-RAS
</td>
<td style="text-align:left;">
DTAI
</td>
<td style="text-align:right;">
0.0153873
</td>
</tr>
<tr>
<td style="text-align:left;">
DTAI-FLT3
</td>
<td style="text-align:left;">
DTAI
</td>
<td style="text-align:right;">
0.0000925
</td>
</tr>
<tr>
<td style="text-align:left;">
DTAI-FLT3-RAS
</td>
<td style="text-align:left;">
DTAI
</td>
<td style="text-align:right;">
0.0327734
</td>
</tr>
<tr>
<td style="text-align:left;">
DTAI-FLT3
</td>
<td style="text-align:left;">
DTAI-RAS
</td>
<td style="text-align:right;">
0.0504716
</td>
</tr>
</tbody>
</table>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:unnamed-chunk-32">Table 3.2: </span>Dominant clone mutations
</caption>
<thead>
<tr>
<th style="text-align:left;">
Group 1
</th>
<th style="text-align:left;">
Group 2
</th>
<th style="text-align:right;">
FDR
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
MPN
</td>
<td style="text-align:left;">
CH
</td>
<td style="text-align:right;">
0.0068538
</td>
</tr>
<tr>
<td style="text-align:left;">
DTAI
</td>
<td style="text-align:left;">
CH
</td>
<td style="text-align:right;">
0.0003515
</td>
</tr>
<tr>
<td style="text-align:left;">
DTAI-RAS
</td>
<td style="text-align:left;">
CH
</td>
<td style="text-align:right;">
0.0003515
</td>
</tr>
<tr>
<td style="text-align:left;">
DTAI-FLT3
</td>
<td style="text-align:left;">
CH
</td>
<td style="text-align:right;">
0.0000133
</td>
</tr>
<tr>
<td style="text-align:left;">
DTAI-FLT3-RAS
</td>
<td style="text-align:left;">
CH
</td>
<td style="text-align:right;">
0.0008256
</td>
</tr>
<tr>
<td style="text-align:left;">
Signaling
</td>
<td style="text-align:left;">
MPN
</td>
<td style="text-align:right;">
0.0620536
</td>
</tr>
<tr>
<td style="text-align:left;">
DTAI-FLT3
</td>
<td style="text-align:left;">
MPN
</td>
<td style="text-align:right;">
0.0620536
</td>
</tr>
<tr>
<td style="text-align:left;">
DTAI
</td>
<td style="text-align:left;">
Signaling
</td>
<td style="text-align:right;">
0.0084549
</td>
</tr>
<tr>
<td style="text-align:left;">
DTAI-RAS
</td>
<td style="text-align:left;">
Signaling
</td>
<td style="text-align:right;">
0.0084549
</td>
</tr>
<tr>
<td style="text-align:left;">
DTAI-FLT3
</td>
<td style="text-align:left;">
Signaling
</td>
<td style="text-align:right;">
0.0003515
</td>
</tr>
<tr>
<td style="text-align:left;">
DTAI-FLT3-RAS
</td>
<td style="text-align:left;">
Signaling
</td>
<td style="text-align:right;">
0.0087669
</td>
</tr>
<tr>
<td style="text-align:left;">
DTAI-FLT3
</td>
<td style="text-align:left;">
DTAI
</td>
<td style="text-align:right;">
0.0602450
</td>
</tr>
<tr>
<td style="text-align:left;">
DTAI-FLT3
</td>
<td style="text-align:left;">
DTAI-RAS
</td>
<td style="text-align:right;">
0.0703147
</td>
</tr>
</tbody>
</table>
<p>A few interesting points in Extended Figure 3</p>
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb54-1"><a href="figure-1-cohort-characterization.html#cb54-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Dominant clone size</span></span>
<span id="cb54-2"><a href="figure-1-cohort-characterization.html#cb54-2" aria-hidden="true" tabindex="-1"></a>gg_dominant_clone_size<span class="ot">&lt;-</span><span class="fu">ggplot</span>(test,</span>
<span id="cb54-3"><a href="figure-1-cohort-characterization.html#cb54-3" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">aes</span>(<span class="at">y=</span>Dominant_clone_size,<span class="at">x=</span>Final_group,<span class="at">fill=</span>Final_group))<span class="sc">+</span></span>
<span id="cb54-4"><a href="figure-1-cohort-characterization.html#cb54-4" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">geom_boxplot</span>(<span class="at">outlier.shape =</span> <span class="cn">NA</span>)<span class="sc">+</span>  </span>
<span id="cb54-5"><a href="figure-1-cohort-characterization.html#cb54-5" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">geom_jitter</span>(<span class="at">width =</span> <span class="fl">0.1</span>,<span class="at">size=</span><span class="fl">0.5</span>)<span class="sc">+</span></span>
<span id="cb54-6"><a href="figure-1-cohort-characterization.html#cb54-6" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">theme_classic</span>(<span class="at">base_size =</span> <span class="dv">8</span>)<span class="sc">+</span></span>
<span id="cb54-7"><a href="figure-1-cohort-characterization.html#cb54-7" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">ylab</span>(<span class="st">&quot;Fraction of sample </span><span class="sc">\n</span><span class="st"> in dominant clone&quot;</span>)<span class="sc">+</span></span>
<span id="cb54-8"><a href="figure-1-cohort-characterization.html#cb54-8" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">xlab</span>(<span class="st">&quot;&quot;</span>)<span class="sc">+</span></span>
<span id="cb54-9"><a href="figure-1-cohort-characterization.html#cb54-9" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle=</span><span class="dv">30</span>,<span class="at">hjust=</span><span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb54-10"><a href="figure-1-cohort-characterization.html#cb54-10" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">scale_fill_brewer</span>(<span class="at">type=</span><span class="st">&quot;seq&quot;</span>,<span class="at">palette =</span> <span class="st">&quot;Reds&quot;</span>,<span class="at">aesthetics =</span> <span class="st">&quot;fill&quot;</span>,<span class="at">guide=</span><span class="cn">FALSE</span>)</span>
<span id="cb54-11"><a href="figure-1-cohort-characterization.html#cb54-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-12"><a href="figure-1-cohort-characterization.html#cb54-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-13"><a href="figure-1-cohort-characterization.html#cb54-13" aria-hidden="true" tabindex="-1"></a><span class="co"># determine the number of mutants alleles in each clone</span></span>
<span id="cb54-14"><a href="figure-1-cohort-characterization.html#cb54-14" aria-hidden="true" tabindex="-1"></a>clone_size_by_genetic_density<span class="ot">&lt;-</span> <span class="fu">do.call</span>(rbind,<span class="fu">lapply</span>(final_sample_summary,<span class="cf">function</span>(x){</span>
<span id="cb54-15"><a href="figure-1-cohort-characterization.html#cb54-15" aria-hidden="true" tabindex="-1"></a>                                    possible_clones_subset <span class="ot">&lt;-</span>x<span class="sc">$</span>Clones<span class="sc">%&gt;%</span><span class="fu">filter</span>(Clone<span class="sc">%in%</span> x<span class="sc">$</span>Clones[,<span class="st">&quot;Clone&quot;</span>] )</span>
<span id="cb54-16"><a href="figure-1-cohort-characterization.html#cb54-16" aria-hidden="true" tabindex="-1"></a>                                    clones<span class="ot">&lt;-</span>possible_clones_subset[,<span class="st">&quot;Clone&quot;</span>]</span>
<span id="cb54-17"><a href="figure-1-cohort-characterization.html#cb54-17" aria-hidden="true" tabindex="-1"></a>                                    dedup<span class="ot">&lt;-</span>x<span class="sc">$</span>NGT[<span class="sc">!</span><span class="fu">duplicated</span>(x<span class="sc">$</span>NGT)<span class="sc">&amp;</span>x<span class="sc">$</span>NGT[,<span class="st">&quot;Clone&quot;</span>]<span class="sc">%in%</span>clones,]</span>
<span id="cb54-18"><a href="figure-1-cohort-characterization.html#cb54-18" aria-hidden="true" tabindex="-1"></a>                                    set_mat<span class="ot">&lt;-</span><span class="fu">full_join</span>(possible_clones_subset[,<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>],dedup)</span>
<span id="cb54-19"><a href="figure-1-cohort-characterization.html#cb54-19" aria-hidden="true" tabindex="-1"></a>                                    counts <span class="ot">&lt;-</span>set_mat[,<span class="st">&quot;Count&quot;</span>]</span>
<span id="cb54-20"><a href="figure-1-cohort-characterization.html#cb54-20" aria-hidden="true" tabindex="-1"></a>                                    weights<span class="ot">&lt;-</span>set_mat[,<span class="st">&quot;Count&quot;</span>]<span class="sc">/</span><span class="fu">sum</span>(set_mat[,<span class="st">&quot;Count&quot;</span>])</span>
<span id="cb54-21"><a href="figure-1-cohort-characterization.html#cb54-21" aria-hidden="true" tabindex="-1"></a>                                    genetic_complexity <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(set_mat[,<span class="sc">-</span><span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>)])</span>
<span id="cb54-22"><a href="figure-1-cohort-characterization.html#cb54-22" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">return</span>(<span class="fu">data.frame</span>(<span class="st">&quot;Clone_size&quot;</span><span class="ot">=</span>weights,</span>
<span id="cb54-23"><a href="figure-1-cohort-characterization.html#cb54-23" aria-hidden="true" tabindex="-1"></a>                                                      <span class="st">&quot;Genetic_density&quot;</span><span class="ot">=</span>genetic_complexity))</span>
<span id="cb54-24"><a href="figure-1-cohort-characterization.html#cb54-24" aria-hidden="true" tabindex="-1"></a>}))</span>
<span id="cb54-25"><a href="figure-1-cohort-characterization.html#cb54-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-26"><a href="figure-1-cohort-characterization.html#cb54-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-27"><a href="figure-1-cohort-characterization.html#cb54-27" aria-hidden="true" tabindex="-1"></a>gg_clone_size_by_genetic_density<span class="ot">&lt;-</span><span class="fu">ggplot</span>(clone_size_by_genetic_density,</span>
<span id="cb54-28"><a href="figure-1-cohort-characterization.html#cb54-28" aria-hidden="true" tabindex="-1"></a>                                              <span class="fu">aes</span>(<span class="at">y=</span>Clone_size,<span class="at">x=</span><span class="fu">factor</span>(Genetic_density),</span>
<span id="cb54-29"><a href="figure-1-cohort-characterization.html#cb54-29" aria-hidden="true" tabindex="-1"></a>                                                  <span class="at">fill=</span><span class="fu">factor</span>(Genetic_density)))<span class="sc">+</span></span>
<span id="cb54-30"><a href="figure-1-cohort-characterization.html#cb54-30" aria-hidden="true" tabindex="-1"></a>                                              <span class="fu">geom_jitter</span>(<span class="at">width =</span> <span class="fl">0.1</span>,<span class="at">size=</span><span class="fl">0.5</span>)<span class="sc">+</span></span>
<span id="cb54-31"><a href="figure-1-cohort-characterization.html#cb54-31" aria-hidden="true" tabindex="-1"></a>                                              <span class="fu">geom_boxplot</span>(<span class="at">outlier.shape =</span> <span class="cn">NA</span>)<span class="sc">+</span>  </span>
<span id="cb54-32"><a href="figure-1-cohort-characterization.html#cb54-32" aria-hidden="true" tabindex="-1"></a>                                              <span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">8</span>)<span class="sc">+</span></span>
<span id="cb54-33"><a href="figure-1-cohort-characterization.html#cb54-33" aria-hidden="true" tabindex="-1"></a>                                              <span class="fu">ylab</span>(<span class="st">&quot;Fraction of sample in clone&quot;</span>)<span class="sc">+</span></span>
<span id="cb54-34"><a href="figure-1-cohort-characterization.html#cb54-34" aria-hidden="true" tabindex="-1"></a>                                              <span class="fu">xlab</span>(<span class="st">&quot;Number of mutant alleles&quot;</span>)<span class="sc">+</span></span>
<span id="cb54-35"><a href="figure-1-cohort-characterization.html#cb54-35" aria-hidden="true" tabindex="-1"></a>                                              <span class="fu">scale_fill_brewer</span>(<span class="at">type=</span><span class="st">&quot;seq&quot;</span>,<span class="at">palette =</span> <span class="st">&quot;Greens&quot;</span>,</span>
<span id="cb54-36"><a href="figure-1-cohort-characterization.html#cb54-36" aria-hidden="true" tabindex="-1"></a>                                                                <span class="at">aesthetics =</span> <span class="st">&quot;fill&quot;</span>,<span class="at">guide=</span><span class="cn">FALSE</span>)</span>
<span id="cb54-37"><a href="figure-1-cohort-characterization.html#cb54-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-38"><a href="figure-1-cohort-characterization.html#cb54-38" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_grid</span>(gg_dominant_clone_size,gg_clone_size_by_genetic_density,<span class="at">align=</span><span class="st">&quot;hv&quot;</span>,<span class="at">axis=</span><span class="st">&quot;tb&quot;</span>,<span class="at">ncol=</span><span class="dv">2</span>,<span class="at">labels=</span><span class="fu">c</span>(<span class="st">&quot;A&quot;</span>,<span class="st">&quot;B&quot;</span>))</span></code></pre></div>
<div class="figure" style="text-align: center"><span id="fig:unnamed-chunk-33"></span>
<img src="scDNA_files/figure-html/unnamed-chunk-33-1.png" alt="Miles et al: Extended Figure 3A-B" width="700.8" />
<p class="caption">
Figure 3.4: Miles et al: Extended Figure 3A-B
</p>
</div>
<div class="sourceCode" id="cb55"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb55-1"><a href="figure-1-cohort-characterization.html#cb55-1" aria-hidden="true" tabindex="-1"></a>pvalues_Dominant_clone_size<span class="ot">&lt;-</span>test<span class="sc">%&gt;%</span>{<span class="fu">melt</span>(<span class="fu">pairwise.t.test</span>(.<span class="sc">$</span>Dominant_clone_size,<span class="at">g=</span>.<span class="sc">$</span>Final_group,</span>
<span id="cb55-2"><a href="figure-1-cohort-characterization.html#cb55-2" aria-hidden="true" tabindex="-1"></a>                                                     <span class="at">data=</span>.,<span class="at">p.adjust.method=</span><span class="st">&quot;fdr&quot;</span>)<span class="sc">$</span>p.value)}<span class="sc">%&gt;%</span></span>
<span id="cb55-3"><a href="figure-1-cohort-characterization.html#cb55-3" aria-hidden="true" tabindex="-1"></a>                                     <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(value))<span class="sc">%&gt;%</span><span class="fu">filter</span>(value<span class="sc">&lt;</span><span class="fl">0.1</span>)</span></code></pre></div>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:unnamed-chunk-35">Table 3.3: </span>Dominant clone size
</caption>
<thead>
<tr>
<th style="text-align:left;">
Group 1
</th>
<th style="text-align:left;">
Group 2
</th>
<th style="text-align:right;">
FDR
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
DTAI-RAS
</td>
<td style="text-align:left;">
CH
</td>
<td style="text-align:right;">
0.0356791
</td>
</tr>
<tr>
<td style="text-align:left;">
DTAI-FLT3
</td>
<td style="text-align:left;">
CH
</td>
<td style="text-align:right;">
0.0123253
</td>
</tr>
<tr>
<td style="text-align:left;">
DTAI-RAS
</td>
<td style="text-align:left;">
MPN
</td>
<td style="text-align:right;">
0.0749752
</td>
</tr>
<tr>
<td style="text-align:left;">
DTAI-FLT3
</td>
<td style="text-align:left;">
MPN
</td>
<td style="text-align:right;">
0.0168044
</td>
</tr>
<tr>
<td style="text-align:left;">
DTAI-RAS
</td>
<td style="text-align:left;">
DTAI
</td>
<td style="text-align:right;">
0.0168044
</td>
</tr>
<tr>
<td style="text-align:left;">
DTAI-FLT3
</td>
<td style="text-align:left;">
DTAI
</td>
<td style="text-align:right;">
0.0081647
</td>
</tr>
</tbody>
</table>

</div>
<div id="clonograph-figure-1d" class="section level3" number="3.1.5">
<h3><span class="header-section-number">3.1.5</span> Clonograph: Figure 1D</h3>
<p>This is probably our favorite way of looking at the data.</p>
<div class="sourceCode" id="cb56"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb56-1"><a href="figure-1-cohort-characterization.html#cb56-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(RColorBrewer)</span>
<span id="cb56-2"><a href="figure-1-cohort-characterization.html#cb56-2" aria-hidden="true" tabindex="-1"></a>final_sample_summary<span class="ot">&lt;-</span><span class="fu">readRDS</span>(<span class="at">file=</span><span class="st">&quot;./data/final_sample_summary.rds&quot;</span>)</span>
<span id="cb56-3"><a href="figure-1-cohort-characterization.html#cb56-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-4"><a href="figure-1-cohort-characterization.html#cb56-4" aria-hidden="true" tabindex="-1"></a>sample <span class="ot">&lt;-</span><span class="st">&quot;MSK8&quot;</span></span>
<span id="cb56-5"><a href="figure-1-cohort-characterization.html#cb56-5" aria-hidden="true" tabindex="-1"></a>sample_list <span class="ot">&lt;-</span>final_sample_summary</span>
<span id="cb56-6"><a href="figure-1-cohort-characterization.html#cb56-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-7"><a href="figure-1-cohort-characterization.html#cb56-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract out the sample of interest    </span></span>
<span id="cb56-8"><a href="figure-1-cohort-characterization.html#cb56-8" aria-hidden="true" tabindex="-1"></a>clonal_abundance <span class="ot">&lt;-</span>sample_list[[sample]]<span class="sc">$</span>Clones </span>
<span id="cb56-9"><a href="figure-1-cohort-characterization.html#cb56-9" aria-hidden="true" tabindex="-1"></a>clonal_architecture <span class="ot">&lt;-</span>sample_list[[sample]]<span class="sc">$</span>Architecture </span>
<span id="cb56-10"><a href="figure-1-cohort-characterization.html#cb56-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-11"><a href="figure-1-cohort-characterization.html#cb56-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Ensure the order of the clone abundance and clone architecture are the same.</span></span>
<span id="cb56-12"><a href="figure-1-cohort-characterization.html#cb56-12" aria-hidden="true" tabindex="-1"></a>clonal_architecture<span class="sc">$</span>Clone <span class="ot">&lt;-</span> <span class="fu">factor</span>(clonal_architecture<span class="sc">$</span>Clone, <span class="at">levels=</span><span class="fu">rev</span>(clonal_abundance<span class="sc">$</span>Clone))</span>
<span id="cb56-13"><a href="figure-1-cohort-characterization.html#cb56-13" aria-hidden="true" tabindex="-1"></a>clonal_abundance<span class="sc">$</span>Clone <span class="ot">&lt;-</span> <span class="fu">factor</span>(clonal_abundance<span class="sc">$</span>Clone, <span class="at">levels=</span><span class="fu">levels</span>(clonal_architecture<span class="sc">$</span>Clone))</span>
<span id="cb56-14"><a href="figure-1-cohort-characterization.html#cb56-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-15"><a href="figure-1-cohort-characterization.html#cb56-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Generate clonal abundance barplot</span></span>
<span id="cb56-16"><a href="figure-1-cohort-characterization.html#cb56-16" aria-hidden="true" tabindex="-1"></a>gg_clonal_barplot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data=</span>clonal_abundance, <span class="fu">aes</span>(<span class="at">x=</span>Clone, <span class="at">y=</span>Count,<span class="at">fill=</span>Count)) <span class="sc">+</span> </span>
<span id="cb56-17"><a href="figure-1-cohort-characterization.html#cb56-17" aria-hidden="true" tabindex="-1"></a>                              <span class="fu">geom_col</span>()<span class="sc">+</span> </span>
<span id="cb56-18"><a href="figure-1-cohort-characterization.html#cb56-18" aria-hidden="true" tabindex="-1"></a>                              <span class="fu">theme_classic</span>(<span class="at">base_size=</span><span class="dv">7</span>)<span class="sc">+</span></span>
<span id="cb56-19"><a href="figure-1-cohort-characterization.html#cb56-19" aria-hidden="true" tabindex="-1"></a>                              <span class="fu">scale_y_continuous</span>(<span class="at">expand=</span><span class="fu">c</span>(<span class="fl">0.01</span>,<span class="dv">0</span>))<span class="sc">+</span></span>
<span id="cb56-20"><a href="figure-1-cohort-characterization.html#cb56-20" aria-hidden="true" tabindex="-1"></a>                              <span class="co">#ylim() + </span></span>
<span id="cb56-21"><a href="figure-1-cohort-characterization.html#cb56-21" aria-hidden="true" tabindex="-1"></a>                              <span class="fu">ylab</span>(<span class="st">&quot;Cell Count&quot;</span>)<span class="sc">+</span></span>
<span id="cb56-22"><a href="figure-1-cohort-characterization.html#cb56-22" aria-hidden="true" tabindex="-1"></a>                              <span class="fu">geom_errorbar</span>(<span class="fu">aes</span>(<span class="at">ymin =</span> LCI, <span class="at">ymax =</span> UCI), <span class="at">width =</span> <span class="fl">0.2</span>)<span class="sc">+</span></span>
<span id="cb56-23"><a href="figure-1-cohort-characterization.html#cb56-23" aria-hidden="true" tabindex="-1"></a>                              <span class="fu">scale_fill_distiller</span>(<span class="at">name =</span> <span class="st">&quot;Value&quot;</span>, <span class="at">palette =</span> <span class="st">&quot;Reds&quot;</span>, <span class="at">direction =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb56-24"><a href="figure-1-cohort-characterization.html#cb56-24" aria-hidden="true" tabindex="-1"></a>                              <span class="fu">theme</span>(<span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(), </span>
<span id="cb56-25"><a href="figure-1-cohort-characterization.html#cb56-25" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">axis.text.x =</span> <span class="fu">element_blank</span>(), </span>
<span id="cb56-26"><a href="figure-1-cohort-characterization.html#cb56-26" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">axis.ticks.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb56-27"><a href="figure-1-cohort-characterization.html#cb56-27" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">axis.line.x =</span><span class="fu">element_blank</span>(),</span>
<span id="cb56-28"><a href="figure-1-cohort-characterization.html#cb56-28" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>,</span>
<span id="cb56-29"><a href="figure-1-cohort-characterization.html#cb56-29" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">plot.margin=</span><span class="fu">unit</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>),<span class="st">&quot;cm&quot;</span>))</span>
<span id="cb56-30"><a href="figure-1-cohort-characterization.html#cb56-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-31"><a href="figure-1-cohort-characterization.html#cb56-31" aria-hidden="true" tabindex="-1"></a><span class="co"># Generate mutation heatmap</span></span>
<span id="cb56-32"><a href="figure-1-cohort-characterization.html#cb56-32" aria-hidden="true" tabindex="-1"></a>gg_heatmap <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data=</span>clonal_architecture,</span>
<span id="cb56-33"><a href="figure-1-cohort-characterization.html#cb56-33" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">aes</span>(<span class="at">x=</span>Clone, <span class="at">y=</span>Mutant, <span class="at">fill=</span>Genotype))<span class="sc">+</span></span>
<span id="cb56-34"><a href="figure-1-cohort-characterization.html#cb56-34" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">geom_tile</span>() <span class="sc">+</span></span>
<span id="cb56-35"><a href="figure-1-cohort-characterization.html#cb56-35" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">scale_fill_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">&quot;WT&quot;</span><span class="ot">=</span><span class="fu">brewer.pal</span>(<span class="dv">7</span>,<span class="st">&quot;Reds&quot;</span>)[<span class="dv">1</span>],</span>
<span id="cb56-36"><a href="figure-1-cohort-characterization.html#cb56-36" aria-hidden="true" tabindex="-1"></a>                                                <span class="st">&quot;Heterozygous&quot;</span><span class="ot">=</span><span class="fu">brewer.pal</span>(<span class="dv">7</span>,<span class="st">&quot;Reds&quot;</span>)[<span class="dv">3</span>],</span>
<span id="cb56-37"><a href="figure-1-cohort-characterization.html#cb56-37" aria-hidden="true" tabindex="-1"></a>                                                <span class="st">&quot;Homozygous&quot;</span><span class="ot">=</span><span class="fu">brewer.pal</span>(<span class="dv">7</span>,<span class="st">&quot;Reds&quot;</span>)[<span class="dv">6</span>],</span>
<span id="cb56-38"><a href="figure-1-cohort-characterization.html#cb56-38" aria-hidden="true" tabindex="-1"></a>                                                <span class="st">&quot;Unknown&quot;</span><span class="ot">=</span><span class="st">&quot;grey50&quot;</span>),<span class="at">name=</span><span class="st">&quot;Genotype&quot;</span>)<span class="sc">+</span></span>
<span id="cb56-39"><a href="figure-1-cohort-characterization.html#cb56-39" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">theme_classic</span>(<span class="at">base_size=</span><span class="dv">7</span>) <span class="sc">+</span></span>
<span id="cb56-40"><a href="figure-1-cohort-characterization.html#cb56-40" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">ylab</span>(<span class="st">&quot;Mutation&quot;</span>)<span class="sc">+</span></span>
<span id="cb56-41"><a href="figure-1-cohort-characterization.html#cb56-41" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">scale_y_discrete</span>(<span class="at">limits =</span> <span class="fu">rev</span>(<span class="fu">levels</span>(clonal_architecture<span class="sc">$</span>Mutant)))<span class="sc">+</span></span>
<span id="cb56-42"><a href="figure-1-cohort-characterization.html#cb56-42" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;right&quot;</span>, <span class="at">legend.direction =</span> <span class="st">&quot;vertical&quot;</span>,</span>
<span id="cb56-43"><a href="figure-1-cohort-characterization.html#cb56-43" aria-hidden="true" tabindex="-1"></a>                          <span class="at">axis.text.x =</span> <span class="fu">element_blank</span>(), </span>
<span id="cb56-44"><a href="figure-1-cohort-characterization.html#cb56-44" aria-hidden="true" tabindex="-1"></a>                          <span class="at">axis.line=</span><span class="fu">element_blank</span>(),</span>
<span id="cb56-45"><a href="figure-1-cohort-characterization.html#cb56-45" aria-hidden="true" tabindex="-1"></a>                          <span class="at">axis.title.x=</span><span class="fu">element_blank</span>(),</span>
<span id="cb56-46"><a href="figure-1-cohort-characterization.html#cb56-46" aria-hidden="true" tabindex="-1"></a>                          <span class="at">axis.ticks.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb56-47"><a href="figure-1-cohort-characterization.html#cb56-47" aria-hidden="true" tabindex="-1"></a>                          <span class="at">plot.margin=</span><span class="fu">unit</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>),<span class="st">&quot;cm&quot;</span>))</span>
<span id="cb56-48"><a href="figure-1-cohort-characterization.html#cb56-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-49"><a href="figure-1-cohort-characterization.html#cb56-49" aria-hidden="true" tabindex="-1"></a><span class="co"># Put it all together</span></span>
<span id="cb56-50"><a href="figure-1-cohort-characterization.html#cb56-50" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_grid</span>(gg_clonal_barplot,gg_heatmap,<span class="at">ncol=</span><span class="dv">1</span>,<span class="at">align=</span><span class="st">&quot;v&quot;</span>,<span class="at">axis=</span><span class="st">&quot;lr&quot;</span>,<span class="at">rel_heights =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="fl">0.75</span>))</span></code></pre></div>
<div class="figure" style="text-align: center"><span id="fig:unnamed-chunk-36"></span>
<img src="scDNA_files/figure-html/unnamed-chunk-36-1.png" alt="Miles et al. Figure 1D" width="576" />
<p class="caption">
Figure 3.5: Miles et al. Figure 1D
</p>
</div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="manuscript-analysis.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="figure-2-clonality.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["scDNA.pdf", "scDNA.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
